---
title: "WR depth per scaffold"
author: "FÃ¡tima S. Barreiro"
date: "April 19, 2020"
output: html_document
---



# Compute the depth of coverage per scaffold (those >13Mbp) and per sample
```{bash}
for bamfile in PATH/TO/BAMFILES/*.bam; do samtools bedcov scaffold_list.bed $bamfile ; done
```




# Visualize depth of coverage per sample, absolute and normalized

Load required libraries
```{r}
library(ggplot2)
library(scales)
library(reshape2)
library(sweep)
library(cowplot)
library(ggpubr)
```

## 1. Choice of scaffolds from the reference assembly based on size

Total length of the assembly
```{r}
# read in scaffold info
fai = read.table("cerSim1.fasta.fai")
scaffolds = as.data.frame(fai[,1:2])

# sort scaffolds by size
ordered_scaffolds = scaffolds[order(scaffolds$V2),]

# calculate cumulative sum of scaffold sizes 
ordered_scaffolds[, 3] <- cumsum(as.numeric(ordered_scaffolds[, 2]))

# total length of assembly
tail(ordered_scaffolds$V3, 1)
```



Calculating the proportion of the genome where we find informative sites
- 63 autosomal scaffolds minus the masked regions
```{r}
1185902722/2464367180
```


Distribution of scaffold sizes
```{r}
Y = ggplot(ordered_scaffolds, aes(V2, V3))
Y = Y + geom_point(colour="royalblue4", size=2, alpha=.8) + theme_bw()
Y = Y + geom_vline(xintercept=1.3e7, colour="darkorange") 
Y = Y + scale_x_continuous(labels = function(x) format(x/1000000, digits=3))
Y = Y + scale_y_continuous(labels = function(x) format(x/1000000, digits=3))
Y = Y + xlab("Scaffold size (Mbp)") + ylab("Cumulative sum of scaffold length (Mbp)")
#Y = Y + theme(axis.text=element_text(size=6, colour="black"), axis.title=element_text(size=7))

Y
```




######################################################################

## 2. Test of depth to identify scaffolds from sex chromosomes among the n = 66 > 13 Mbp

Read in files
```{r}
# retrieve files
Depthfiles66 <- read.table('WR_56I_DepthScafNew.txt', header=F)
```

```{r}
# initialize empty dataframe to collect the data and the new columns
B = NULL
```

Expand the data frame with the pertinent calculations
```{r}
for (i in levels(Depthfiles66$V5)) {
  
  # read in files
  b <- Depthfiles66[Depthfiles66$V5 == i, ]
  colnames(b) = c("Scaffold", "InitialPosition", "EndPosition", "DepthCounts", "Sample")
  
  # compute total depth of coverage per individual
  TotalDepthInd = sum(as.numeric(b[,4]))
  
  # calculate the total length of the scaffolds under consideration
  totLen = sum(as.numeric(b[,3]))
  
  # compute average depth of coverage per individual
  b$AveDepthInd = TotalDepthInd/totLen

  # compute average depth of coverage per scaffold
  b$AveScaffoldDepth = b$DepthCounts/b$EndPosition
  
  # Calculate the normalised depth per scaffold by dividing scaffold depth by total depth
  b$NormScaffoldDepth = b$AveScaffoldDepth / b$AveDepthInd

  # store all these subtables into a full data frame
  B <- rbind(B, b)
}

head(B)
```

Make an x axis
```{r}
scfIndex = as.data.frame(seq(1, 66, 1))
colnames(scfIndex) = c("scfIndex")
```



Visualize the normalized depths across scaffolds and per sample
```{r}
Z = ggplot(B[B$Sample != 'mwr_TG0457' & B$Sample != 'mwr_TG0459' & B$Sample != 'mwr_TG0464' & B$Sample != 'mwr_TGF08', ], aes(Scaffold, NormScaffoldDepth, colour=Sample)) +  geom_point(size=1, alpha=.6) +
  theme_bw() + guides(colour=F) 

Z = Z + scale_y_continuous(limits=c(0,2)) + scale_x_discrete(labels=scfIndex$scfIndex, guide = guide_axis(n.dodge = 2))
Z = Z + ylab("Normalised depth of coverage")

Z
```



Save scaffold-related plots as a supplementary figure

```{r}
# pdf('WR_v5_FigS2.pdf', useDingbats = F)
#  plot_grid(Y, Z, nrow=2, ncol=1, labels=c('A', 'B'), label_size = 10, align="v")
# dev.off()
```



##############################################################################

## 3. Depth of coverage per sample across the final chosen scaffolds (n = 63)


Read in files
```{r}
# retrieve files
Depthfiles <- read.table('WR_56I_DepthScafNew_63scfs.txt', header=F)
```

```{r}
# initialize empty dataframe to collect the data and the new columns
A = NULL
```


Calculate depth per individual
```{r}
for (i in levels(Depthfiles$V5)) {
  
  # read in files
  a <- Depthfiles[Depthfiles$V5 == i, ]
  colnames(a) = c("Scaffold", "InitialPosition", "EndPosition", "DepthCounts", "Sample")
  
  # compute total depth of coverage per individual
  TotalDepthInd = sum(as.numeric(a[,4]))
  
  # calculate the total length of the scaffolds under consideration
  totLen = sum(as.numeric(a[,3]))
  
  # compute average depth of coverage per individual
  a$AveDepthInd = TotalDepthInd/totLen

  # compute average depth of coverage per scaffold
  a$AveScaffoldDepth = a$DepthCounts/a$EndPosition

  # store all these subtables into a full data frame
  A <- rbind(A, a)
}

head(A)

```


##############################################################################################

#### Plot per subspecies

Subset the dataframe of depths into the two subspecies
```{r}
NWR_names = c("mwr_TG0450", "mwr_TG0451", "mwr_TG0452", "mwr_TG0453", "mwr_TG0454", "mwr_TG0455", "mwr_TG0456", "mwr_TG0458", "mwr_TG0460", "mwr_TG0461", "mwr_TG0462", "mwr_TG0463", "mwr_TG0465", "mwr_TGF14", "mwr_TGF15", "mwr_TGF16", "wr_SRR5852741", "wr_SRR5852742", "wr_SRR5852743", "wr_SRR5852744", "wr_SRR5852745", "wr_SRR5852746", "wr_SRR5852747", "wr_SRR5852748", "wr_SRR5852749")

NWR = A[A$Sample %in% NWR_names,]

SWR_names = c("mwr_TG0449", "mwr_TGF01", "mwr_TGF02", "mwr_TGF03", "mwr_TGF04", "mwr_TGF05", "mwr_TGF06", "mwr_TGF07", "mwr_TGF09", "mwr_TGF10", "mwr_TGF11", "mwr_TGF12", "mwr_TGF13", "mwr_TGF17", "wr_P9109_102", "wr_P9109_103", "wr_P9109_104", "wr_P9109_105", "wr_P9109_106", "wr_P9109_107", "wr_P9109_108", "wr_P9109_109", "wr_P9109_110", "wr_SRR5852738", "wr_SRR5852739", "wr_SRR5852740", "wr_SRR5852750")

SWR = A[A$Sample %in% SWR_names,]

```


Set the new labels
```{r}
NWR_newID = c('CD-un.1','CD1912.1','CD1912.2','CD1911.1','CD1913.1','CD1912.3','CD1911.2','CD-un.2','SD1905.1','SD1905.2','UG1905.1','SD1905.3','SD1905.4','SD1914.1','UG1905.2','SD1905.5', 'SD1972.1','SD1973.1','SS1954.1','SD1972.2','SD1974.1','SD1972.3','SD1963.1','UG1965.1','SD1972.4')

SWR_newID = c('ZA1869.1','ZA1845.1','un1856.1','ZA-un.1','ZA1845.2','NA1856.1','NA1845.1','NA1857.1','ZA1842.1','ZA1960.1','ZA1965.1','ZA1970.1','un1968.1','ZA1976.1','cap2009.1','cap2010.1','cap2008.1','cap2011.1','cap2011.2','cap2011.3','ZA2012.1','ZA1968.1','ZA1969.1','ZA1958.1','ZA1963.1','ZA1963.2','ZA1961.1')
  
```


Add a column for time window to make a colour legend when plotting
```{r}
NWR$time_window <- "window"

NWR[NWR$Sample %in% c("mwr_TG0450", "mwr_TG0451", "mwr_TG0452", "mwr_TG0453", "mwr_TG0454", "mwr_TG0455", "mwr_TG0456", "mwr_TG0458", "mwr_TG0460", "mwr_TG0461", "mwr_TG0462", "mwr_TG0463", "mwr_TG0465", "mwr_TGF14", "mwr_TGF15", "mwr_TGF16"), ]$time_window <- "NWRpre"

NWR[NWR$Sample %in% c("wr_SRR5852741", "wr_SRR5852742", "wr_SRR5852743", "wr_SRR5852744", "wr_SRR5852745", "wr_SRR5852746", "wr_SRR5852747", "wr_SRR5852748", "wr_SRR5852749") , ]$time_window <- "NWRpost"

```


```{r}
SWR$time_window <- "window"

SWR[SWR$Sample %in% c("mwr_TG0449", "mwr_TGF01", "mwr_TGF02", "mwr_TGF03", "mwr_TGF04", "mwr_TGF05", "mwr_TGF06", "mwr_TGF07", "mwr_TGF09"), ]$time_window <- "SWRpre"

SWR[SWR$Sample %in% c("mwr_TGF10", "mwr_TGF11", "mwr_TGF12", "mwr_TGF13", "mwr_TGF17", "wr_P9109_102", "wr_P9109_103", "wr_P9109_104", "wr_P9109_105", "wr_P9109_106", "wr_P9109_107", "wr_P9109_108", "wr_P9109_109", "wr_P9109_110", "wr_SRR5852738", "wr_SRR5852739", "wr_SRR5852740", "wr_SRR5852750") , ]$time_window <- "SWRpost"

```




Plot NWR
```{r}
NWR$time_window=factor(NWR$time_window, levels=c("NWRpre","NWRpost"))
```

```{r}
n = ggplot(NWR, aes(Sample, AveScaffoldDepth)) 

n = n + 
  theme_bw() + theme(legend.position="none", axis.text.x=element_text(angle=40, hjust=1, size=8)) +
  geom_violin(aes(Sample, AveScaffoldDepth, colour=time_window), fill="gray60", alpha=.4, lwd=.5, adjust=5) +
  scale_colour_manual(values=c("gold3", "darkorange")) +
  scale_x_discrete(labels=as.character(NWR_newID)) +
  scale_y_continuous(limits=c(0, 25)) +
  stat_summary(fun="median", geom="crossbar", width=.09, lwd=.2, colour="gray20") +
  labs(x="Sample name", y="Depth of coverage")

n
```




Plot SWR1
```{r}
SWR$time_window=factor(SWR$time_window, levels=c("SWRpre","SWRpost"))
```

```{r}
s = ggplot(SWR, aes(Sample, AveScaffoldDepth)) 

s = s + 
  theme_bw() + theme(legend.position = "none", axis.text.x=element_text(angle=40, hjust=1, size=8)) +
  geom_violin(aes(Sample, AveScaffoldDepth, colour=time_window), fill="gray60", alpha=.5, lwd=.4, adjust=5) +
  scale_colour_manual(values=c("royalblue4", "steelblue3")) +
  scale_x_discrete(labels=as.character(SWR_newID)) +
  scale_y_continuous(limits=c(0, 25)) +
  stat_summary(fun="median", geom="crossbar", width=.09, lwd=.2, colour="gray20") +
  labs(x="Sample name", y="Depth of coverage")

s
```



```{r}
figS2v5 = ggarrange(n + theme(axis.title.x=element_blank()) , s , nrow=2, align="v", labels=c("A", "B"), font.label=list(size=10))
figS2v5
```


Save as supplementary figure
```{r}
# pdf('WR_v5_FigS1_v2.pdf', useDingbats = F)
# figS2v5
# dev.off()

```

































