---
title: "WR Error Rates"
author: "FÃ¡tima S. Barreiro"
date: "October 8, 2020"
output: html_document
---



## Estimating error rates for individual bam files with ANGSD
```{bash}
# index the fasta files you want to use as best sample and outgroup in the Fastas folder
samtools faidx /PATH/TO/BEST/SAMPLE/wr_P9109_110.fa
samtools faidx /PATH/TO/OUTGROUP/SAMPLE/OP5012.fa

# Error estimation using an outgroup (black rhino OP5012) and an error free individual (modern white rhino wr_SRR5852749)
module load angsd/0.921
angsd -doAncError 1 -anc /PATH/TO/OUTGROUP/SAMPLE/OP5012.fa -ref /PATH/TO/BEST/SAMPLE/wr_P9109_110.fa -out WR_ErrorRates_P9109_110best_BRout.Est -bam WR_52I_bamlist.txt
```




## Visualization of results

Load required libraries
```{r message=FALSE, warning=FALSE, paged.print=FALSE}
library(ggplot2)
library(cowplot)
library(ggrepel)
library(reshape2)
library(dplyr)
library(tidyverse)
library(data.table)
require(scales)
library(ggpubr)
library(tidyr)
library(PairedData)
library(gridExtra)
```


Reading in file containing metadata and genomic diversity metrics 
```{r}
data = read.table("WR_ErrorRates_P9109_110best_BRout.Est.txt", header=T, fill=T, stringsAsFactors=T, fileEncoding="latin1")
```


Reading in file containing metadata and genomic diversity metrics 
```{r}
metadata = read.csv("WR_TableS1_v6.csv", h=T, fill=T, stringsAsFactors=T, fileEncoding="latin1")
```



### Visualizing the error rates per sample

Bind the sample lab_IDs to the error rates dataframe so each row of error rates can be assigned to its corresponding sample
```{r}
# subset the error rate rows of interest
smalldata = data[1:52,]

# take from the metadata the relevant identification columns
IDs = as.data.frame(metadata$lab_ID)
colnames(IDs) = "ID"

samples = as.data.frame(metadata$sample_name)
colnames(samples) = "sample_name"

window = as.data.frame(metadata$time_window)
colnames(window) = "time_window"


# bind the error rate subsetted data with the identifier columns
smalldata = cbind(smalldata, IDs, samples, window)
head(smalldata)
```


Melt the data to facilitate plotting
```{r}
smalldataM = reshape2::melt(smalldata)
head(smalldataM)
```



Reorder the entries so that the samples appeared grouped by time window on the plot
```{r}
smalldataM$time_window=factor(smalldataM$time_window,levels=c("NWRpre", "NWRpost", "SWRpre", "SWRpost"))
```


Visualizing the error rates (all types) per sample

```{r}
E = ggplot(smalldataM, aes(sample_name, value, fill=variable, colour=variable)) + 
  geom_bar(position = position_dodge(0.6), stat = "identity", width=0.4, size=0.2) + theme_bw() +
  theme(axis.text.x=element_text(angle=80, hjust=1, size=8), legend.key.size= unit(4, "mm"), legend.title=element_blank()) +
  ylab("Error rate") + xlab("Sample name") +
  facet_wrap(~time_window, scales = "free_x")

E
```



Save the plot
```{r eval=FALSE, include=FALSE}
pdf("WR_v5_FigS3_ErrorRates.pdf", height=5, useDingbats = F)
E
dev.off()
```




### Calculate a transversion composite error rate with which to correct the Het estimates to buffer the effect of the SRR batch

Calculate a composite error rate at Tv sites by summing up all error rates per row
```{r}
smalldata %>% 
  dplyr::select(GtoT, TtoG, AtoT, TtoA, AtoC, CtoA) %>% 
  rowSums(na.rm=TRUE) -> smalldata$ErTv

head(smalldata)
```


Save these error rates (single and composite) as a separate file 
```{r}
write.table(smalldata, file="WR_Errors_ErTv.csv", row.names=F, col.names=T, sep=",")
```




### Calculate the corrected het by operating with the composite error rate and the het estimate

```{r}
smalldataHc = cbind(smalldataH, smalldata$ErTv)
head(smalldataHc)

# subtract from the Het estimate the excess driven by transversion error rates
smalldataHc = smalldataHc %>% 
  mutate(HET3x_corrected = HET3x-(HET3x*smalldata$ErTv), HET5x_corrected = HET5x-(HET5x*smalldata$ErTv) )

```



Visualise the new het estimates
```{r}
tinydataHc = smalldataHc %>%
  dplyr::select(ID, HET3x, HET5x, HET3x_corrected, HET5x_corrected)

tinydataHc = reshape2::melt(tinydataHc)
```


```{r}
P = ggplot(tinydataHc, aes(ID, value, colour=variable)) + geom_point() + theme_bw() +
  theme(axis.text.x=element_text(angle=80, hjust=1), legend.title = element_blank(), legend.position = "top")
  
P
```





### Correlations between error rates and estimates of heterozygosity

Parse the dataframe: add the heterozygosity estimates from the metadata. In this case I used the estimates with GL2 and filtering sites below 3x
```{r}
HET3x = metadata$gw_het_GL2_3x
HET5x = metadata$gw_het_GL2_5x

smalldataH = cbind(smalldata, HET3x, HET5x)
head(smalldataH)
```



Create a function to run and visualize linear regressions
```{r}

correlate_vars = function(dataframe, het_var, error_var, ID) {
  plot = ggscatter(dataframe, x = het_var, y = error_var, label = ID,
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson")
  return(plot)
  print(plot)
}

```


Use the function to visualise correlation plots of every error rate with the two Het estimates chosen
```{r}
correlate_vars(smalldataH, "HET3x", "TtoC", "ID")
correlate_vars(smalldataH, "HET3x", "CtoT", "ID")
correlate_vars(smalldataH, "HET3x", "GtoA", "ID")
correlate_vars(smalldataH, "HET3x", "AtoG", "ID")
correlate_vars(smalldataH, "HET3x", "GtoT", "ID")
correlate_vars(smalldataH, "HET3x", "TtoG", "ID")
correlate_vars(smalldataH, "HET3x", "AtoT", "ID")
correlate_vars(smalldataH, "HET3x", "TtoA", "ID")
correlate_vars(smalldataH, "HET3x", "GtoC", "ID")
correlate_vars(smalldataH, "HET3x", "CtoG", "ID")
correlate_vars(smalldataH, "HET3x", "AtoC", "ID")
correlate_vars(smalldataH, "HET3x", "CtoA", "ID")

```



```{r}
correlate_vars(smalldataH, "HET5x", "TtoC", "ID")
correlate_vars(smalldataH, "HET5x", "CtoT", "ID")
correlate_vars(smalldataH, "HET5x", "GtoA", "ID")
correlate_vars(smalldataH, "HET5x", "AtoG", "ID")
correlate_vars(smalldataH, "HET5x", "GtoT", "ID")
correlate_vars(smalldataH, "HET5x", "TtoG", "ID")
correlate_vars(smalldataH, "HET5x", "AtoT", "ID")
correlate_vars(smalldataH, "HET5x", "TtoA", "ID")
correlate_vars(smalldataH, "HET5x", "GtoC", "ID")
correlate_vars(smalldataH, "HET5x", "CtoG", "ID")
correlate_vars(smalldataH, "HET5x", "AtoC", "ID")
correlate_vars(smalldataH, "HET5x", "CtoA", "ID")

```


















