---
title: "WR_indDiv_deltaEstimators"
author: "FÃ¡tima S. Barreiro"
date: "January 30, 2020"
output: html_document
---
```{r}

```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

Load required libraries
```{r include=FALSE}
library(ggplot2)
library(cowplot)
library(ggrepel)
library(reshape2)
library(dplyr)
library(tidyverse)
library(data.table)
require(scales)
library(ggpubr)
library(tidyr)
library(PairedData)
library(gridExtra)
```



# Analyses and visualization of white rhino genomic data

## 1. Nucleotide diversity based on transversion sites per group 
Read in data
```{r}
pi_NWR = read.csv("NWR_nucDiversity_tv.csv", header=T)
pi_SWR = read.csv("SWR_nucDiversity_tv.csv", header=T)
```

Bind relevant columns and reshape
```{r}
pi_WR = as.data.frame(cbind(pi_NWR$pi_NWR1, pi_NWR$pi_NWR2, pi_SWR$pi_SWR1, pi_SWR$pi_SWR2))
colnames(pi_WR) = c("NWR1", "NWR2", "SWR1", "SWR2")

pi_WR = reshape2::melt(pi_WR)

```

Summary statistics for pi_tv across the 4 groups
```{r}
pi_WR %>%
  group_by(variable) %>%
  summarize(min=min(value), max=max(value), median=median(value))
```


Statistical comparisons of medians between pi-tv in pre- and post-bottleneck samples per subspecies
```{r}
pi_nwr1 = subset(pi_WR,  variable == "NWR1", value, drop = TRUE)
pi_nwr2 = subset(pi_WR,  variable == "NWR2", value, drop = TRUE)
pi_swr1 = subset(pi_WR,  variable == "SWR1", value, drop = TRUE)
pi_swr2 = subset(pi_WR,  variable == "SWR2", value, drop = TRUE)


wilcox.test(pi_nwr1, pi_nwr2, paired=TRUE)
wilcox.test(pi_swr1, pi_swr2, paired=TRUE)
```


Visualization
```{r}
# plot
ggplot(pi_WR, aes(pi_WR$variable, pi_WR$value, fill=pi_WR$variable))  +
  
  # format scatter points
  geom_jitter(aes(colour=pi_WR$variable), shape=17, position=position_jitter(0.11), cex=.7, alpha=.8) +  
  
  # format boxplots
  geom_boxplot(alpha=.2, lwd=.2, fill='gray') +
  scale_color_manual(values= (c("gold2", "darkorange", "royalblue4", "steelblue3"))) +
  
  # get summary stats and groups comparison
  stat_compare_means(paired=TRUE, method = "wilcox.test", comparisons= list(c("NWR1", "NWR2")), label="p.format", label.y=0.00063, size=1.7) +
  stat_compare_means(paired=TRUE, method = "wilcox.test", comparisons= list(c("SWR1", "SWR2")), label="p.format", label.y=0.00042, size=1.7) +
  
  # embellish
  scale_y_continuous(limits=c(0.0001,0.00070), labels = function(x) format(x, scientific = FALSE)) +
  labs(x=NULL, y="Nucleotide diversity-tv per scaffold") + 
  theme_bw() + theme(legend.position = "none", axis.text.x=element_text(size=6), axis.title.y=element_text(size=6), axis.text.y=element_text(size=5))
  

```




## 2. Individual measures of genomic diversity: Statistical comparisons between time windows and delta estimators 

Reading in file containing metadata and genomic diversity metrics 
```{r}
data = read.csv("WR_TableS2_v2.csv", h=T, fill=T, stringsAsFactors=T, fileEncoding="latin1")
```


Subsetting data per group (subspecies and time window)
```{r}
nwr1 = data[data$time_window == "NWR1",]
nwr2 = data[data$time_window == "NWR2",]
swr1 = data[data$time_window == "SWR1",]
swr2 = data[data$time_window == "SWR2",]
```


### 2.a) Genome-wide heterozygosity

Summary statistics for gw_het across the 4 groups
```{r}
data %>%
  group_by(time_window) %>%
  summarize(min=min(corrected_gw_het), max=max(corrected_gw_het), median=median(corrected_gw_het))
```


Statistical comparison of gw_het median between pre- and post-bottleneck samples for each subspecies
```{r}
wilcox.test(nwr1$corrected_gw_het, nwr2$corrected_gw_het, paired = F)
wilcox.test(swr1$corrected_gw_het, swr2$corrected_gw_het, paired = F)
```


Visualization
```{r}
# plot het values per time window
ggplot(data, aes(time_window, corrected_gw_het)) + 
  
  # plot violin per group
  geom_violin(alpha=.2, aes(colour=time_window, fill=time_window),  lwd=.2) +

  # add scatter points per group
  geom_jitter(aes(colour=time_window), shape=16, position=position_jitter(0.11), cex=.6, alpha=.8) + 
  
  # add colours and filling colours
  scale_fill_manual(values= (c("gold2", "darkorange", "royalblue4", "steelblue3"))) + 
  scale_color_manual(values= (c("gold2", "darkorange", "royalblue4", "steelblue3"))) +
  
  # add stats and an unpaired wilcox test to compare the medians
  stat_summary(fun.y=mean, fun.ymin = function (corrected_gw_het) mean(corrected_gw_het) - sd(corrected_gw_het), 
               fun.ymax = function(corrected_gw_het) mean(corrected_gw_het) + sd(corrected_gw_het), geom = "pointrange" , size=.12, stroke=.4, col="black") +
  stat_summary(fun.y=median, geom="point", col="black", shape=2, size=1) +
  
  stat_compare_means(paired=F, method = "wilcox.test", comparisons= list(c("NWR1", "NWR2"), c("SWR1", "SWR2")), label.y=c(0.0005, 0.0003), size=1.7) +
  
  # embellish
  scale_y_continuous(limits=c(0, 0.0005),labels = function(x) format(x, scientific = FALSE)) +
  theme_bw() + theme(legend.position = "none", axis.text.x=element_text(size=6), axis.title.y=element_text(size=6), axis.text.y=element_text(size=5)) +
  labs(x=NULL, y="Genome-wide heterozygosity-tv") 
```



### 2.b) Exon heterozygosity

Summary statistics for exon_het across the 4 groups
```{r}
data %>%
  group_by(time_window) %>%
  summarize(min=min(corrected_exon_het), max=max(corrected_exon_het), median=median(corrected_exon_het))
```


Statistical comparison of exon_het median between pre- and post-bottleneck samples for each subspecies
```{r}
wilcox.test(nwr1$corrected_exon_het, nwr2$corrected_exon_het, paired = F)
wilcox.test(swr1$corrected_exon_het, swr2$corrected_exon_het, paired = F)
```


Visualization
```{r}
ggplot(data, aes(time_window, corrected_exon_het, fill=time_window)) + 
  
  # plot violin per group
  geom_violin(alpha=.2, aes(colour=time_window, fill=time_window),  lwd=.2) +

  # add scatter points per group
  geom_jitter(aes(colour=time_window), shape=16, position=position_jitter(0.11), cex=.6, alpha=.8) + 
  
  # stats for exon violin
  stat_summary(fun.y=mean, fun.ymin = function (corrected_exon_het) mean(corrected_exon_het) - sd(corrected_exon_het), 
               fun.ymax = function(corrected_exon_het) mean(corrected_exon_het) + sd(corrected_exon_het), geom = "pointrange" , size=.12, stroke=.4) +
  stat_summary(fun.y=median, geom="point", col="black", shape=2, size=1) +
  
  # compare medians between violins
  stat_compare_means(paired=F, method = "wilcox.test", comparisons= list(c("NWR1", "NWR2"), c("SWR1", "SWR2")), label.y=c(0.00036, 0.00025), size=1.7) +
  
  # choose colours
  scale_fill_manual(values= (c("gold2", "darkorange", "royalblue4", "steelblue3"))) + 
  scale_color_manual(values= (c("gold2", "darkorange", "royalblue4", "steelblue3"))) +
  
  # embellish
  scale_y_continuous(limits=c(0, 0.0005), labels = function(x) format(x, scientific = FALSE)) +
  theme_bw() + theme(legend.position = "none", axis.text.x=element_text(size=6), axis.title.y=element_text(size=6), axis.text.y=element_text(size=5)) +
  labs(x=NULL, y="Exon heterozygosity-tv")
  
```



### 2.c) RoH length and FRoH

Read in data on RoH number and length for each sample and reshape appropriately
```{r}

# read in the file to determine the max number of columns
x <- scan("WR_RoH_perscaffold.txt", what = "", sep = "\n")
x <- strsplit(x, "[ \t]+") # split string by white space
max.col <- max(sapply(x, length))

# specify col.names as ?read.table suggests and set old header to missing values
cn <- paste("RoH", 1:max.col, sep = "")
roh <- read.table("WR_RoH_perscaffold.txt", fill = TRUE, col.names = cn, na.strings = "RoH")

# delete old header (first row)
roh = roh[-1,]

```


Calculate length of each RoH per sample
```{r}
# calculate length of RoH per sample in bp
roh_len = roh*1000000 - ((roh-1)*500000)
```


Add relevant metadata from the 'data' dataframe
```{r}
# bind relevant metadata to the RoHs length
new_ID = data$new_ID
country = data$Country
time_window = data$time_window

roh_len_info = cbind(new_ID, country, time_window, roh_len)

# melt data with metadata variables as ID
roh_len_info = reshape2::melt(roh_len_info, id.vars = c("new_ID", "country", "time_window"))
```



Visualize distribution of heterozygosity values per sample
```{r}
a=ggplot(roh_len_info[time_window=='NWR1',], aes(value, colour=new_ID)) +
  geom_density(aes(y=5e5 * ..count..)) +
  scale_x_continuous(limits=c(1e6, 3e7), breaks=c(1e6, 5e6, 1e7, 1.5e7, 2e7, 2.5e7), labels=c(1, 5, 10, 15, 20, 25)) + 
  theme_bw() +
  theme(legend.title = element_blank(), legend.text = element_text(size = 5), 
        legend.position = c(.9, 0.5), legend.background = element_rect(fill=NA, size=0.1, linetype = 0),
        legend.key.size = unit(.3, "cm")) +
  labs(x="RoH length (Mbp)", y ="Number of RoH")

b=ggplot(roh_len_info[time_window=='NWR2',], aes(value, colour=new_ID)) +
  geom_density(aes(y=5e5 * ..count..)) +
  scale_x_continuous(limits=c(1e6, 3e7), breaks=c(1e6, 5e6, 1e7, 1.5e7, 2e7, 2.5e7), labels=c(1, 5, 10, 15, 20, 25)) + 
  theme_bw() +
  theme(legend.title = element_blank(), legend.text = element_text(size = 5), 
        legend.position = c(.9, 0.5), legend.background = element_rect(fill=NA, size=0.1, linetype = 0),
        legend.key.size = unit(.3, "cm")) +
  labs(x="RoH length (Mbp)", y ="Number of RoH")

c=ggplot(roh_len_info[time_window=='SWR1',], aes(value, colour=new_ID)) +
  geom_density(aes(y=5e5 * ..count..)) +
  scale_x_continuous(limits=c(1e6, 3e7), breaks=c(1e6, 5e6, 1e7, 1.5e7, 2e7, 2.5e7), labels=c(1, 5, 10, 15, 20, 25)) + 
  theme_bw() +
  theme(legend.title = element_blank(), legend.text = element_text(size = 5), 
        legend.position = c(.9, 0.5), legend.background = element_rect(fill=NA, size=0.1, linetype = 0),
        legend.key.size = unit(.3, "cm")) +
  labs(x="RoH length (Mbp)", y ="Number of RoH")

d=ggplot(roh_len_info[time_window=='SWR2',], aes(value, colour=new_ID)) +
  geom_density(aes(y=5e5 * ..count..)) +
  scale_x_continuous(limits=c(1e6, 3e7), breaks=c(1e6, 5e6, 1e7, 1.5e7, 2e7, 2.5e7), labels=c(1, 5, 10, 15, 20, 25)) + 
  theme_bw() +
  theme(legend.title = element_blank(), legend.text = element_text(size = 5), 
        legend.position = c(.9, 0.5), legend.background = element_rect(fill=NA, size=0.1, linetype = 0),
        legend.key.size = unit(.3, "cm")) +
  labs(x="RoH length (Mbp)", y ="Number of RoH")


plot_grid(a, b, c, d, labels=c("NWR1", "NWR2", "SWR1", "SWR2"), label_size=7, ncol = 2, nrow = 2)

```


Based on these plots we chose 5e-5 as the threshold of low heterozygosity to flag a long-enough region as a RoH. Also we excluded after this we ran the upcoming tests for both the whole sample set and for a set without 3 outliers driven by missing data due to depth of coverage: CD-un.1, CD1911.2, ZA1845.1


Prepare a data frame like roh_len_info without the outliers
```{r}
# filtering 3 outlier samples out
roh_len_info_filt = roh_len_info[roh_len_info$new_ID != 'CD-un.1' & roh_len_info$new_ID != 'CD1911.2' & roh_len_info$new_ID != 'ZA1845.1', ]
```


Filter out isolated windows of low heterozygosity (not considered RoH)
```{r}
roh_len_info_above1Mb = roh_len_info[roh_len_info$value > 1000000,]
roh_len_info_filt_above1Mb = roh_len_info_filt[roh_len_info_filt$value > 1000000,]
```


Calculate summary statistics for RoH length to include in plots for both the entire sample set and without the 3 outliers
```{r}
# stats for all samples
group = roh_len_info_above1Mb %>%
  group_by(time_window) %>%
  summarize(min=round((min(value)/1000000), digits=2), max=round((max(value)/1000000), digits=2), 
            mean=round((mean(value)/1000000), digits=2))

# stats for filtered samples (-3 outliers)
group_filt = roh_len_info_filt_above1Mb %>%
  group_by(time_window) %>%
  summarize(min=round((min(value)/1000000), digits=2), max=round((max(value)/1000000), digits=2), 
            mean=round((mean(value)/1000000), digits=2))

```



Visualize RoH length distrubution per group for all samples and for the filtered sample set
```{r}
A = ggplot(roh_len_info_above1Mb, aes(value, colour=time_window, fill=time_window)) +
  geom_density(aes(y=5e5 * ..count..), alpha=.08, lwd=.4) +
  scale_colour_manual(values= (c("gold2", "darkorange", "royalblue4", "steelblue3"))) +
  scale_fill_manual(values= (c("gold2", "darkorange", "royalblue4", "steelblue3"))) +
  scale_x_continuous(limits=c(1.5e6, 22e6), breaks=c(1.5e6, 5e6, 10e6, 15e6, 20e6), 
                     labels=c(1.5, 5, 10, 15, 20)) + 
  theme_bw() +
  theme(legend.title = element_blank(), legend.text = element_text(size=5), 
        legend.position = c(0.78, 0.9), legend.background = element_rect(fill=NA, size=0.1, linetype = 0),
        legend.key.size = unit(.35, "cm"), axis.title=element_text(size=7), axis.text=element_text(size=7)) +
  labs(x="RoH length (Mbp)", y ="Number of RoH") +
  annotation_custom(tableGrob(group_filt[1:4,], rows=NULL, theme = ttheme_default(base_size=5)), 
                    xmin=13e6, xmax=17e6, ymin=200, ymax=400)

B = ggplot(roh_len_info_filt_above1Mb, aes(value, colour=time_window, fill=time_window)) +
  geom_density(aes(y=5e5 * ..count..), alpha=.08, lwd=.4) +
  scale_colour_manual(values= (c("gold2", "darkorange", "royalblue4", "steelblue3"))) +
  scale_fill_manual(values= (c("gold2", "darkorange", "royalblue4", "steelblue3"))) +
  scale_x_continuous(limits=c(1.5e6, 22e6), breaks=c(1.5e6, 5e6, 10e6, 15e6, 20e6), 
                     labels=c(1.5, 5, 10, 15, 20)) + 
  theme_bw() +
  theme(legend.title = element_blank(), legend.text = element_text(size=5), 
        legend.position = c(0.78, 0.9), legend.background = element_rect(fill=NA, size=0.1, linetype = 0),
        legend.key.size = unit(.35, "cm"), axis.title=element_text(size=7), axis.text=element_text(size=7)) +
  labs(x="RoH length (Mbp)", y ="Number of RoH") +
  annotation_custom(tableGrob(group_filt[1:4,], rows=NULL, theme = ttheme_default(base_size=5)), 
                    xmin=13e6, xmax=17e6, ymin=200, ymax=400)

plot_grid(A, B, labels=c('unfiltered dataset (n=52)', 'filtered dataset (n=49)'), label_size=9, ncol = 2, nrow = 1)

```


Calculate FRoH as an estimate of inbreeding by dividing total length falling in RoH by the total genomic length scanned for each sample
```{r}
# transpose and assign sample IDs as column names
roh_f = t(roh_len)
colnames(roh_f) = data$new_ID

# turn values equal or lower to 1Mbp into missing data
roh_f[roh_f<=1000000]=NA

# sum all remaning values per column and divide by total length of windows screened
# the total length screened is the number of windows multiplied by 1Mbp, minus 
# number of windows minus 1 multiplied by 0.5 Mbp ->
# -> (3659 windows*1Mbp) - ((3659-1)*0.5Mbp)

Froh = as.data.frame(colSums(roh_f, na.rm=T) / 1830000000)
colnames(Froh) = "froh"

# add metadata
new_ID = data$new_ID
country = data$Country
time_window = data$time_window

# create dataframes for all samples and for filtered set
Froh_f = cbind(new_ID, country, time_window, Froh)
Froh_f_filt = Froh_f[Froh_f$new_ID != 'CD-un.1' & Froh_f$new_ID != 'CD1911.2' & Froh_f$new_ID != 'ZA1845.1', ]

```


Get summary statistics for Froh across groups for both the unfiltered and filtered sample sets
```{r}
Froh_f %>%
  group_by(time_window) %>%
  summarize(min=min(froh, na.rm = TRUE), max=max(froh, na.rm = TRUE), median = median(froh, na.rm = TRUE))

Froh_f_filt %>%
  group_by(time_window) %>%
  summarize(min=min(froh, na.rm = TRUE), max=max(froh, na.rm = TRUE), median = median(froh, na.rm = TRUE))
```


Statistical comparison of Froh median between pre- and post-bottleneck samples for each subspecies for the unfiltered data (reported only in supplementary material)
```{r}
# data subsetting
nwr1 = Froh_f[Froh_f$time_window == "NWR1",]
nwr2 = Froh_f[Froh_f$time_window == "NWR2",]
swr1 = Froh_f[Froh_f$time_window == "SWR1",]
swr2 = Froh_f[Froh_f$time_window == "SWR2",]

# calculating medians
median(nwr1$froh)
median(nwr2$froh)
median(swr1$froh)
median(swr2$froh)

# comparing medians of Froh between pre- and post-bottleneck
wilcox.test(nwr1$froh, nwr2$froh, paired = F)
wilcox.test(swr1$froh, swr2$froh, paired = F)
```


Statistical comparison of Froh median between pre- and post-bottleneck samples for each subspecies for the filtered data
```{r}
# data subsetting
nwr1 = Froh_f_filt[Froh_f_filt$time_window == "NWR1",]
nwr2 = Froh_f_filt[Froh_f_filt$time_window == "NWR2",]
swr1 = Froh_f_filt[Froh_f_filt$time_window == "SWR1",]
swr2 = Froh_f_filt[Froh_f_filt$time_window == "SWR2",]

# calculating medians
median(nwr1$froh)
median(nwr2$froh)
median(swr1$froh)
median(swr2$froh)

# comparing medians of Froh between pre- and post-bottleneck
wilcox.test(nwr1$froh, nwr2$froh, paired = F)
wilcox.test(swr1$froh, swr2$froh, paired = F)
```


Visualize Froh across groups
```{r}
# plot violins per group with stats for unfiltered sample set
A = ggplot(Froh_f, aes(time_window, froh, fill=time_window)) +
  geom_violin(alpha=.3, aes(colour=time_window), lwd=.2) +
  
  # add scatter points per group
  geom_jitter(aes(colour=time_window), shape=16, position=position_jitter(0.11), cex=1, alpha=.9) + 
  
  # add colours and filling colours
  scale_fill_manual(values= (c("gold2", "darkorange", "royalblue4", "steelblue3"))) + 
  scale_color_manual(values= (c("gold2", "darkorange", "royalblue4", "steelblue3"))) +
  
  # add stats
  stat_summary(fun.y=mean, size=0.14, fun.ymin = function (froh) mean(froh) - sd(froh), 
               fun.ymax = function(froh) mean(froh) + sd(froh), geom = "pointrange", stroke=.5) +
  stat_summary(fun.y=median, size=1.8, geom="point", col="black", shape=2) +
  
  stat_compare_means(paired=F, method = "wilcox.test", comparisons= list(c("NWR1", "NWR2"), c("SWR1", "SWR2")), 
                     label.y=c(0.41, 0.53), size=3) +
  
  # embellish
  scale_y_continuous(limits=c(0, 0.6), labels = function(x) format(x, scientific = FALSE)) +
  theme_bw() + theme(legend.position = "none", axis.title=element_text(size=7), axis.text=element_text(size=7)) +
  labs(x="Time window", y="F-RoH")

# plot violins per group with stats for filtered sample set
B = ggplot(Froh_f_filt, aes(time_window, froh, fill=time_window)) +
  geom_violin(alpha=.3, aes(colour=time_window), lwd=.2) +
  
  # add scatter points per group
  geom_jitter(aes(colour=time_window), shape=16, position=position_jitter(0.11), cex=1, alpha=.9) + 
  
  # add colours and filling colours
  scale_fill_manual(values= (c("gold2", "darkorange", "royalblue4", "steelblue3"))) + 
  scale_color_manual(values= (c("gold2", "darkorange", "royalblue4", "steelblue3"))) +
  
  # add stats
  stat_summary(fun.y=mean, size=0.14, fun.ymin = function (froh) mean(froh) - sd(froh), 
               fun.ymax = function(froh) mean(froh) + sd(froh), geom = "pointrange", stroke=.5) +
  stat_summary(fun.y=median, size=1.8, geom="point", col="black", shape=2) +
  
  stat_compare_means(paired=F, method = "wilcox.test", comparisons= list(c("NWR1", "NWR2"), c("SWR1", "SWR2")), 
                     label.y=c(0.41, 0.53), size=3) +
  
  # embellish
  scale_y_continuous(limits=c(0, 0.6), labels = function(x) format(x, scientific = FALSE)) +
  theme_bw() + theme(legend.position = "none", axis.title=element_text(size=7), axis.text=element_text(size=7)) +
  labs(x="Time window", y="F-RoH")

# combine plots in one page
plot_grid(A, B, labels=c('unfiltered dataset (n=52)', 'filtered dataset (n=49)'), label_size=9, nrow=1 , ncol=2)
```




## 3. Individual diversity metrics in subsampled dataset

Subsampling the full dataset to assess patterns of genomic erosion after mitigated sampling bias
Basically, from NWR1 we exclude samples originating from DRC because that genetic origin is not represented in our set of NWR2
From SWR2 we exclude captive individuals, since they were bred in zoos, so we focus only on wild SWR when comparing pre- and post-bottleneck

```{r}
data_ss = subset(data, Country != 'DRC' & Country != 'captive' | is.na(Country))
```

Subsetting data per group (subspecies and time window)
```{r}
nwr1_ss = data_ss[data_ss$time_window == "NWR1",]
nwr2_ss = data_ss[data_ss$time_window == "NWR2",]
swr1_ss = data_ss[data_ss$time_window == "SWR1",]
swr2_ss = data_ss[data_ss$time_window == "SWR2",]
```


### 3.a) Genome-wide heterozygosity in subsampled 

Summary statistics for gw_het across the 4 groups
```{r}
data_ss %>%
  group_by(time_window) %>%
  summarize(min=min(corrected_gw_het), max=max(corrected_gw_het), median=median(corrected_gw_het))
```


Statistical comparison of gw_het median between pre- and post-bottleneck samples for each subspecies
```{r}
wilcox.test(nwr1_ss$corrected_gw_het, nwr2_ss$corrected_gw_het, paired = F)
wilcox.test(swr1_ss$corrected_gw_het, swr2_ss$corrected_gw_het, paired = F)
```


Visualization
```{r}
# plot het values per time window
ggplot(data_ss, aes(time_window, corrected_gw_het)) + 
  
  # plot violin per group
  geom_violin(alpha=.2, aes(colour=time_window, fill=time_window),  lwd=.2) +

  # add scatter points per group
  geom_jitter(aes(colour=time_window), shape=16, position=position_jitter(0.11), cex=.6, alpha=.8) + 
  
  # add colours and filling colours
  scale_fill_manual(values= (c("gold2", "darkorange", "royalblue4", "steelblue3"))) + 
  scale_color_manual(values= (c("gold2", "darkorange", "royalblue4", "steelblue3"))) +
  
  # add stats and an unpaired wilcox test to compare the medians
  stat_summary(fun.y=mean, fun.ymin = function (corrected_gw_het) mean(corrected_gw_het) - sd(corrected_gw_het), 
               fun.ymax = function(corrected_gw_het) mean(corrected_gw_het) + sd(corrected_gw_het), geom = "pointrange" , size=.12, stroke=.4, col="black") +
  stat_summary(fun.y=median, geom="point", col="black", shape=2, size=1) +
  
  stat_compare_means(paired=F, method = "wilcox.test", comparisons= list(c("NWR1", "NWR2"), c("SWR1", "SWR2")), label.y=c(0.0005, 0.0003), size=1.7) +
  
  # embellish
  scale_y_continuous(limits=c(0, 0.0005),labels = function(x) format(x, scientific = FALSE)) +
  theme_bw() + theme(legend.position = "none", axis.text.x=element_text(size=6), axis.title.y=element_text(size=6), axis.text.y=element_text(size=5)) +
  labs(x=NULL, y="Genome-wide heterozygosity-tv") 
```



### 3.b) Exon heterozygosity in subsampled data

Summary statistics for exon_het across the 4 groups
```{r}
data_ss %>%
  group_by(time_window) %>%
  summarize(min=min(corrected_exon_het), max=max(corrected_exon_het), median=median(corrected_exon_het))
```

Statistical comparison of exon_het median between pre- and post-bottleneck samples for each subspecies
```{r}
wilcox.test(nwr1_ss$corrected_exon_het, nwr2_ss$corrected_exon_het, paired = F)
wilcox.test(swr1_ss$corrected_exon_het, swr2_ss$corrected_exon_het, paired = F)
```

Visualization
```{r}
ggplot(data_ss, aes(time_window, corrected_exon_het, fill=time_window)) + 
  
  # plot violin per group
  geom_violin(alpha=.2, aes(colour=time_window, fill=time_window),  lwd=.2) +

  # add scatter points per group
  geom_jitter(aes(colour=time_window), shape=16, position=position_jitter(0.11), cex=.6, alpha=.8) + 
  
  # stats for exon violin
  stat_summary(fun.y=mean, fun.ymin = function (corrected_exon_het) mean(corrected_exon_het) - sd(corrected_exon_het), 
               fun.ymax = function(corrected_exon_het) mean(corrected_exon_het) + sd(corrected_exon_het), geom = "pointrange" , size=.12, stroke=.4) +
  stat_summary(fun.y=median, geom="point", col="black", shape=2, size=1) +
  
  # compare medians between violins
  stat_compare_means(paired=F, method = "wilcox.test", comparisons= list(c("NWR1", "NWR2"), c("SWR1", "SWR2")), label.y=c(0.00036, 0.00025), size=1.7) +
  
  # choose colours
  scale_fill_manual(values= (c("gold2", "darkorange", "royalblue4", "steelblue3"))) + 
  scale_color_manual(values= (c("gold2", "darkorange", "royalblue4", "steelblue3"))) +
  
  # embellish
  scale_y_continuous(limits=c(0, 0.0005), labels = function(x) format(x, scientific = FALSE)) +
  theme_bw() + theme(legend.position = "none", axis.text.x=element_text(size=6), axis.title.y=element_text(size=6), axis.text.y=element_text(size=5)) +
  labs(x=NULL, y="Exon heterozygosity-tv")
  
```



### 3.c) RoH length and FRoH in subsampled data

Two out of the three outlier samples removed in RoH analyses, orginate from DRC, so they would be exluded anyway from the calculations below
That would leave only one out of three outliers in the subsampled dataset, so we chose to perform these calculations only on the filtered dataset, without any of the 3 outliers
Subsampling by country is therefore an added filtering step on top of removing the 3 outliers (n = 37 final number of samples)

Take already parsed data for RoH length after removing outliers and filter out pertinent samples (from DRC and captive) 
```{r}
roh_len_info_filt_above1Mb_ss = subset(roh_len_info_filt_above1Mb, country != 'DRC' & country != 'captive' | is.na(country))
```


Calculate summary statistics for RoH length to include in plots for both the entire sample set and without the 3 outliers
```{r}
group_filt_ss = roh_len_info_filt_above1Mb_ss %>%
  group_by(time_window) %>%
  summarize(min=round((min(value)/1000000), digits=2), max=round((max(value)/1000000), digits=2), 
            mean=round((mean(value)/1000000), digits=2))

```


Visualize RoH length distrubution per group for all samples and for the filtered sample set
```{r}
ggplot(roh_len_info_filt_above1Mb_ss, aes(value, colour=time_window, fill=time_window)) +
  geom_density(aes(y=5e5 * ..count..), alpha=.08, lwd=.4) +
  scale_colour_manual(values= (c("gold2", "darkorange", "royalblue4", "steelblue3"))) +
  scale_fill_manual(values= (c("gold2", "darkorange", "royalblue4", "steelblue3"))) +
  scale_x_continuous(limits=c(1.5e6, 22e6), breaks=c(1.5e6, 5e6, 10e6, 15e6, 20e6), 
                     labels=c(1.5, 5, 10, 15, 20)) + 
  theme_bw() +
  theme(legend.title = element_blank(), legend.text = element_text(size=5), 
        legend.position = c(0.78, 0.9), legend.background = element_rect(fill=NA, size=0.1, linetype = 0),
        legend.key.size = unit(.35, "cm"), axis.title=element_text(size=7), axis.text=element_text(size=7)) +
  labs(x="RoH length (Mbp)", y ="Number of RoH", title="RoH length distribution in subsampled dataset") +
  annotation_custom(tableGrob(group_filt[1:4,], rows=NULL, theme = ttheme_default(base_size=5)), 
                    xmin=13e6, xmax=17e6, ymin=200, ymax=400)

```


Take the relevant values of FRoH, those corresponding to the samples present in the subsampled dataset
```{r}
Froh_f_filt_ss = subset(Froh_f_filt, country != 'DRC' & country != 'captive' | is.na(country))
```


Get summary statistics for FRoH across groups for the subsampled dataset without outliers (n = 37)
```{r}
Froh_f_filt_ss %>%
  group_by(time_window) %>%
  summarize(min=min(froh, na.rm = TRUE), max=max(froh, na.rm = TRUE), median = median(froh, na.rm = TRUE))
```


Statistical comparison of Froh median between pre- and post-bottleneck samples for each subspecies for the subsampled data (n = 37)
```{r}
# data subsetting
nwr1 = Froh_f_filt_ss[Froh_f_filt_ss$time_window == "NWR1",]
nwr2 = Froh_f_filt_ss[Froh_f_filt_ss$time_window == "NWR2",]
swr1 = Froh_f_filt_ss[Froh_f_filt_ss$time_window == "SWR1",]
swr2 = Froh_f_filt_ss[Froh_f_filt_ss$time_window == "SWR2",]

# comparing medians of Froh between pre- and post-bottleneck
wilcox.test(nwr1$froh, nwr2$froh, paired = F)
wilcox.test(swr1$froh, swr2$froh, paired = F)
```

Visualize Froh across groups
```{r}
# plot violins per group with stats for filtered sample set
ggplot(Froh_f_filt_ss, aes(time_window, froh, fill=time_window)) +
  geom_violin(alpha=.3, aes(colour=time_window), lwd=.2) +
  
  # add scatter points per group
  geom_jitter(aes(colour=time_window), shape=16, position=position_jitter(0.11), cex=1, alpha=.9) + 
  
  # add colours and filling colours
  scale_fill_manual(values= (c("gold2", "darkorange", "royalblue4", "steelblue3"))) + 
  scale_color_manual(values= (c("gold2", "darkorange", "royalblue4", "steelblue3"))) +
  
  # add stats
  stat_summary(fun.y=mean, size=0.14, fun.ymin = function (froh) mean(froh) - sd(froh), 
               fun.ymax = function(froh) mean(froh) + sd(froh), geom = "pointrange", stroke=.5) +
  stat_summary(fun.y=median, size=1.8, geom="point", col="black", shape=2) +
  
  stat_compare_means(paired=F, method = "wilcox.test", comparisons= list(c("NWR1", "NWR2"), c("SWR1", "SWR2")), 
                     label.y=c(0.41, 0.53), size=3) +
  
  # embellish
  scale_y_continuous(limits=c(0, 0.6), labels = function(x) format(x, scientific = FALSE)) +
  theme_bw() + theme(legend.position = "none", axis.title=element_text(size=7), axis.text=element_text(size=7)) +
  labs(x="Time window", y="F-RoH")
```









