---
title: "White rhino genomic erosion, code for calculations and figures"
author: "FÃ¡tima S. Barreiro"
date: "January 30, 2020"
output: html_document
---

# Analyses and visualization of group, and individual-level metrics of genomic diversity


Load required libraries
```{r message=FALSE, warning=FALSE, paged.print=FALSE}
library(ggplot2)
library(cowplot)
library(ggrepel)
library(reshape2)
library(dplyr)
library(tidyverse)
library(data.table)
require(scales)
library(ggpubr)
library(tidyr)
library(PairedData)
library(gridExtra)
```




## 1. Nucleotide diversity based on transversion sites per group 

Read in data
```{r include=FALSE}
pi_NWR = read.csv("NWR_nucDiversity_tv.csv", header=T)
pi_SWR = read.csv("SWR_nucDiversity_tv.csv", header=T)
```

Bind relevant columns and reshape
```{r}
pi_WR = as.data.frame(cbind(pi_NWR$pi_NWR1, pi_NWR$pi_NWR2, pi_SWR$pi_SWR1, pi_SWR$pi_SWR2))
colnames(pi_WR) = c("NWR1", "NWR2", "SWR1", "SWR2")

pi_WR = reshape2::melt(pi_WR)

```

Summary statistics for pi_tv across the 4 groups
```{r}
pi_WR %>%
  group_by(variable) %>%
  summarize(min=min(value), max=max(value), median=median(value))
```


Statistical comparisons of medians between pi-tv in pre- and post-bottleneck samples per subspecies
```{r}
pi_nwr1 = subset(pi_WR,  variable == "NWR1", value, drop = TRUE)
pi_nwr2 = subset(pi_WR,  variable == "NWR2", value, drop = TRUE)
pi_swr1 = subset(pi_WR,  variable == "SWR1", value, drop = TRUE)
pi_swr2 = subset(pi_WR,  variable == "SWR2", value, drop = TRUE)


wilcox.test(pi_nwr1, pi_nwr2, paired=TRUE)
wilcox.test(pi_swr1, pi_swr2, paired=TRUE)
```


Visualization
```{r}
# plot
pi = ggplot(pi_WR, aes(variable, value, fill=variable))  +
   theme_bw() + theme(legend.position = "none", axis.text=element_text(size=7, colour='gray10'), axis.title.y=element_text(size=7), 
                     panel.grid.major=(element_line(size=.12, colour='gray80')))
  
  # format scatter points
pi = pi +
    # format boxplots
  geom_boxplot(alpha=.2, lwd=.2, fill='gray30') +
  scale_color_manual(values= (c("gold3", "darkorange", "royalblue4", "steelblue3"))) +
  # add the points
  geom_jitter(aes(colour=variable), shape=17, position=position_jitter(0.11), cex=.7, alpha=.7) +  
  
  # get summary stats and groups comparison
  stat_compare_means(paired=TRUE, method = "wilcox.test", comparisons= list(c("NWR1", "NWR2")), label="p.format", label.y=0.00064, size=2.5) +
  stat_compare_means(paired=TRUE, method = "wilcox.test", comparisons= list(c("SWR1", "SWR2")), label="p.format", label.y=0.00044, size=2.5) +
  
  # embellish
  scale_y_continuous(limits=c(0.0001,0.00068), labels = function(x) format(x, scientific = FALSE)) +
  labs(x=NULL, y="Nucleotide diversity (tv)") + 
  theme_bw() + theme(legend.position = "none", axis.text=element_text(size=7, colour='gray10'), axis.title.y=element_text(size=7), 
                     panel.grid.major=(element_line(size=.12, colour='gray80'))) 
  
pi
```

####################################################
Missing: Pi-tv for subsampled dataset (running in computerome)
Did not work* 
################################################






## 2. Individual measures of genomic diversity: Statistical comparisons between time windows and delta estimators 


Reading in file containing metadata and genomic diversity metrics 
```{r}
data = read.csv("WR_v3_TableS1_v1.csv", h=T, fill=T, stringsAsFactors=T, fileEncoding="latin1")
```


Subsetting data per group (subspecies and time window)
```{r}
nwr1 = data[data$time_window == "NWR1",]
nwr2 = data[data$time_window == "NWR2",]
swr1 = data[data$time_window == "SWR1",]
swr2 = data[data$time_window == "SWR2",]
```



### 2.a) Genome-wide heterozygosity

Summary statistics for gw_het across the 4 groups
```{r}
data %>%
  group_by(time_window) %>%
  summarize(min=min(corrected_gw_het), max=max(corrected_gw_het), median=median(corrected_gw_het))
```


Statistical comparison of gw_het median between pre- and post-bottleneck samples for each subspecies
```{r}
wilcox.test(nwr1$corrected_gw_het, nwr2$corrected_gw_het, paired = F)
wilcox.test(swr1$corrected_gw_het, swr2$corrected_gw_het, paired = F)
```


Visualization
```{r}
# plot het values per time window
gw_het = ggplot(data, aes(time_window, corrected_gw_het)) + 
  theme_bw() + theme(legend.position = "none", axis.text=element_text(size=7, colour='gray10'), axis.title.y=element_text(size=7), 
                     panel.grid.major=(element_line(size=.12, colour='gray80')))
  
  
      # plot violin per group
gw_het = gw_het +  geom_violin(aes(colour=time_window, fill=time_window), lwd=.3, alpha=.4) +
  
  # add scatter points per group
  geom_jitter(colour='gray40', shape=16, position=position_jitter(0.11), cex=.6, alpha=.7) + 
  
  # add colours and filling colours
  scale_fill_manual(values= (c("gold3", "darkorange", "royalblue4", "steelblue3"))) + 
  scale_color_manual(values= (c("gold3", "darkorange", "royalblue4", "steelblue3"))) +
  
  # add stats and an unpaired wilcox test to compare the medians
  stat_summary(fun=median, geom="crossbar", colour='black', width=.12, lwd=.3) +
  
  stat_summary(fun=mean, fun.min = function (corrected_gw_het) mean(corrected_gw_het) - sd(corrected_gw_het), 
               fun.max = function(corrected_gw_het) mean(corrected_gw_het) + sd(corrected_gw_het), geom = "pointrange" , size=.12, stroke=.4, col="black") +
  
  stat_compare_means(paired=F, method = "wilcox.test", comparisons= list(c("NWR1", "NWR2"), c("SWR1", "SWR2")), label.y=c(0.00048, 0.00022), size=2.5) +
  
  # embellish
  scale_y_continuous(limits=c(0.00007, 0.0005),labels = function(x) format(x, scientific = FALSE)) +
  
  
  labs(x=NULL, y="Genome-wide heterozygosity (tv)")

gw_het
```



### 2.b) Exon heterozygosity

Summary statistics for exon_het across the 4 groups
```{r}
data %>%
  group_by(time_window) %>%
  summarize(min=min(corrected_exon_het), max=max(corrected_exon_het), median=median(corrected_exon_het))
```


Statistical comparison of exon_het median between pre- and post-bottleneck samples for each subspecies
```{r}
wilcox.test(nwr1$corrected_exon_het, nwr2$corrected_exon_het, paired = F)
wilcox.test(swr1$corrected_exon_het, swr2$corrected_exon_het, paired = F)
```


Visualization
```{r}
exon_het = ggplot(data, aes(time_window, corrected_exon_het, fill=time_window)) + 
  theme_bw() + theme(legend.position = "none", axis.text=element_text(size=7, colour='gray10'), axis.title.y=element_text(size=7), 
                     panel.grid.major=(element_line(size=.12, colour='gray80')))

    # plot violin per group
exon_het = exon_het +  geom_violin(aes(colour=time_window, fill=time_window), lwd=.3, alpha=.4) +
  
  # add scatter points per group
  geom_jitter(colour='gray40', shape=16, position=position_jitter(0.11), cex=.6, alpha=.7) + 
  
  # add colours and filling colours
  scale_fill_manual(values= (c("gold3", "darkorange", "royalblue4", "steelblue3"))) + 
  scale_color_manual(values= (c("gold3", "darkorange", "royalblue4", "steelblue3"))) +
  
  # add stats and an unpaired wilcox test to compare the medians
  stat_summary(fun=median, geom="crossbar", colour='black', width=.12, lwd=.3) +
  
  stat_summary(fun=mean, fun.min = function (corrected_gw_het) mean(corrected_gw_het) - sd(corrected_gw_het), 
               fun.max = function(corrected_gw_het) mean(corrected_gw_het) + sd(corrected_gw_het), geom = "pointrange" , size=.12, stroke=.4, col="black") +
  
  stat_compare_means(paired=F, method = "wilcox.test", comparisons= list(c("NWR1", "NWR2"), c("SWR1", "SWR2")), label.y=c(0.00033, 0.00022), size=2.5) +
  
  # embellish
  scale_y_continuous(limits=c(0.00004, 0.00035),labels = function(x) format(x, scientific = FALSE)) +
  
  
  labs(x=NULL, y="Exon heterozygosity (tv)")

exon_het
  
```


#### Make genomic erosion figure (Figure 3 in WR v3)

First make the panel of heterozygosities
```{r}
HET = plot_grid(gw_het, exon_het, nrow=1, ncol=2, labels=c('B','C'), label_size=7)
HET
```



```{r}
pdf('WR_v3_Fig3_v1.pdf', height=5, width=5, useDingbats = F)
plot_grid(pi, HET, nrow=2, ncol=1, labels=c('A', NULL), label_size = 7, rel_heights = c(0.7, 1.2))
dev.off()
```




#####################################################################################


## 3. Individual diversity metrics in subsampled dataset


Subsampling the full dataset to assess patterns of genomic erosion after mitigated sampling bias
Basically, from NWR1 we exclude samples originating from DRC because that genetic origin is not represented in our set of NWR2
From SWR2 we exclude captive individuals, since they were bred in zoos, so we focus only on wild SWR when comparing pre- and post-bottleneck

```{r}
data_ss = subset(data, country != 'DRC' & country != 'captive' | is.na(country))
```

Subsetting data per group (subspecies and time window)
```{r}
nwr1_ss = data_ss[data_ss$time_window == "NWR1",]
nwr2_ss = data_ss[data_ss$time_window == "NWR2",]
swr1_ss = data_ss[data_ss$time_window == "SWR1",]
swr2_ss = data_ss[data_ss$time_window == "SWR2",]
```



### 3.a) Genome-wide heterozygosity in subsampled dataset


Summary statistics for gw_het across the 4 groups
```{r}
data_ss %>%
  group_by(time_window) %>%
  summarize(min=min(corrected_gw_het), max=max(corrected_gw_het), median=median(corrected_gw_het))
```


Statistical comparison of gw_het median between pre- and post-bottleneck samples for each subspecies
```{r}
wilcox.test(nwr1_ss$corrected_gw_het, nwr2_ss$corrected_gw_het, paired = F)
wilcox.test(swr1_ss$corrected_gw_het, swr2_ss$corrected_gw_het, paired = F)
```


Visualization
```{r}
# plot het values per time window
gw_het_ss = ggplot(data_ss, aes(time_window, corrected_gw_het)) + 
  theme_bw() + theme(legend.position = "none", axis.text=element_text(size=7, colour='gray10'), axis.title.y=element_text(size=7), 
                     panel.grid.major=(element_line(size=.12, colour='gray80')))

    # plot violin per group
gw_het_ss = gw_het_ss +  geom_violin(aes(colour=time_window, fill=time_window), lwd=.3, alpha=.4) +
  
  # add scatter points per group
  geom_jitter(colour='gray40', shape=16, position=position_jitter(0.11), cex=.6, alpha=.7) + 
  
  # add colours and filling colours
  scale_fill_manual(values= (c("gold3", "darkorange", "royalblue4", "steelblue3"))) + 
  scale_color_manual(values= (c("gold3", "darkorange", "royalblue4", "steelblue3"))) +
  
  # add stats and an unpaired wilcox test to compare the medians
  stat_summary(fun=median, geom="crossbar", colour='black', width=.12, lwd=.3) +
  
  stat_summary(fun=mean, fun.min = function (corrected_gw_het) mean(corrected_gw_het) - sd(corrected_gw_het), 
               fun.max = function(corrected_gw_het) mean(corrected_gw_het) + sd(corrected_gw_het), geom = "pointrange" , size=.12, stroke=.4, col="black") +
  
  stat_compare_means(paired=F, method = "wilcox.test", comparisons= list(c("NWR1", "NWR2"), c("SWR1", "SWR2")), label.y=c(0.00023, 0.00022), size=2.5) +
  
  # embellish
  scale_y_continuous(limits=c(0.00008, 0.00025),labels = function(x) format(x, scientific = FALSE)) +
  
  
  labs(x=NULL, y="Genome-wide heterozygosity (tv)")

gw_het_ss
```



### 3.b) Exon heterozygosity in subsampled data


Summary statistics for exon_het across the 4 groups
```{r}
data_ss %>%
  group_by(time_window) %>%
  summarize(min=min(corrected_exon_het), max=max(corrected_exon_het), median=median(corrected_exon_het))
```

Statistical comparison of exon_het median between pre- and post-bottleneck samples for each subspecies
```{r}
wilcox.test(nwr1_ss$corrected_exon_het, nwr2_ss$corrected_exon_het, paired = F)
wilcox.test(swr1_ss$corrected_exon_het, swr2_ss$corrected_exon_het, paired = F)
```

Visualization
```{r}
exon_het_ss = ggplot(data_ss, aes(time_window, corrected_exon_het, fill=time_window)) + 
    theme_bw() + theme(legend.position = "none", axis.text=element_text(size=7, colour='gray10'), axis.title.y=element_text(size=7), 
                     panel.grid.major=(element_line(size=.12, colour='gray80')))

    # plot violin per group
exon_het_ss = exon_het_ss +  geom_violin(aes(colour=time_window, fill=time_window), lwd=.3, alpha=.4) +
  
  # add scatter points per group
  geom_jitter(colour='gray40', shape=16, position=position_jitter(0.11), cex=.6, alpha=.7) + 
  
  # add colours and filling colours
  scale_fill_manual(values= (c("gold3", "darkorange", "royalblue4", "steelblue3"))) + 
  scale_color_manual(values= (c("gold3", "darkorange", "royalblue4", "steelblue3"))) +
  
  # add stats and an unpaired wilcox test to compare the medians
  stat_summary(fun=median, geom="crossbar", colour='black', width=.12, lwd=.3) +
  
  stat_summary(fun=mean, fun.min = function (corrected_gw_het) mean(corrected_gw_het) - sd(corrected_gw_het), 
               fun.max = function(corrected_gw_het) mean(corrected_gw_het) + sd(corrected_gw_het), geom = "pointrange" , size=.12, stroke=.4, col="black") +
  
  stat_compare_means(paired=F, method = "wilcox.test", comparisons= list(c("NWR1", "NWR2"), c("SWR1", "SWR2")), label.y=c(0.00014, 0.00014), size=2.5) +
  
  # embellish
  scale_y_continuous(limits=c(0.00005, 0.00015),labels = function(x) format(x, scientific = FALSE)) +
  
  
  labs(x=NULL, y="Exon heterozygosity (tv)")

exon_het_ss
```



#### Make Figure on individual genomic erosion for supplementary 
```{r}
pdf('WR_v3_FigS8.pdf', height=4, width=5, useDingbats = F)
plot_grid(gw_het_ss, exon_het_ss, nrow=1, ncol=2, labels=c('A','B'), label_size=7)
dev.off()
```





######################################### Hetwindows, RoH length and FRoH #####################################################


## 4. Heterozygosity distribution across windows (1Mbp, 0.5 Mbp slide)

Visualize distribution of heterozygosity values per sample
```{r}
# read het data across 52 inds
hets = read.table("WR_hetwindows_info.hets", header=T)

## change headers in order to use new IDs instead of the old ones
# get only columns corresponding to sample values
hets_noinfo = hets[,4:55]
# get new id labels from metadata
metadata <-read.csv("/home/fatima/Documents/PhD/COMPUTATIONAL/metadata/WR_PaleomixStats_metadata_final.csv", h=T, fill=T, stringsAsFactors=T, fileEncoding="latin1")
new_ID = metadata$new_ID
# change column names on data
colnames(hets_noinfo) = new_ID
hets_newID = cbind(hets[,1:3], hets_noinfo)

## reshape dataframes with old ID and new ID to then get the names of the old IDs as an extra column
df_oldID = reshape2::melt(hets, id.vars=c("scaffold", "midpoint", "window"))
setnames(df_oldID, "variable", "oldID")

df_newID = melt(hets_newID, id.vars=c("scaffold", "midpoint", "window"))

# rename the extra old ID column as oldID only
df = cbind(df_newID, df_oldID$oldID)
setnames(df, "df_oldID$oldID", "oldID")
```


Make subsets of the data per group
```{r}
## subset data to avoid collapsing R due to huge data frame
NWR1_names = c("mwr_TG0450", "mwr_TG0451", "mwr_TG0452", "mwr_TG0453", "mwr_TG0454", "mwr_TG0455", "mwr_TG0456", "mwr_TG0458", "mwr_TG0460", "mwr_TG0461", "mwr_TG0462", "mwr_TG0463", "mwr_TG0465", "mwr_TGF14", "mwr_TGF15", "mwr_TGF16")
NWR1 = df[df$oldID %in% NWR1_names,]

NWR2_names = c("wr_SRR5852741", "wr_SRR5852742", "wr_SRR5852743", "wr_SRR5852744", "wr_SRR5852745", "wr_SRR5852746", "wr_SRR5852747", "wr_SRR5852748", "wr_SRR5852749")
NWR2 = df[df$oldID %in% NWR2_names,]

SWR1_names = c("mwr_TG0449", "mwr_TGF01", "mwr_TGF02", "mwr_TGF03", "mwr_TGF04", "mwr_TGF05", "mwr_TGF06", "mwr_TGF07", "mwr_TGF09")
SWR1 = df[df$oldID %in% SWR1_names,]

SWR2_names = c("mwr_TGF10", "mwr_TGF11", "mwr_TGF12", "mwr_TGF13", "mwr_TGF17", "wr_P9109_102", "wr_P9109_103", "wr_P9109_104", "wr_P9109_105", "wr_P9109_106", "wr_P9109_107", "wr_P9109_108", "wr_P9109_109", "wr_P9109_110", "wr_SRR5852738", "wr_SRR5852739", "wr_SRR5852740", "wr_SRR5852750")
SWR2 = df[df$oldID %in% SWR2_names,]

```


Visualize the distribution of local estimates of heterozygosity per sample across windows in order to determine the threshold of heterozygosity below which long stretches will be flagged as RoH
```{r}
ggplot(NWR1, aes(variable, value, colour=variable)) + 
  # filter(roh_len_info[time_window=='NWR1',], !is.na(new_ID)) %>% 
  # ggplot(aes(new_ID, value)) +
  geom_violin(aes(variable, value, colour=variable, fill=variable), alpha=.3, lwd=.3) +
  theme_bw() +
  scale_y_continuous(limits=c(0, 1e-3)) +
  geom_hline(yintercept=5e-5, color="grey20", linetype="dashed", size=.4) +
  theme_bw() + 
  theme(legend.position = "none", axis.title=element_text(size=5), axis.text.x=element_text(size=4, angle=20, hjust=1), axis.text.y=element_text(size=5)) +
  labs(x=NULL, y="Heterozygosity per 1Mbp window")

ggplot(NWR2, aes(variable, value, colour=variable)) + 
  # filter(roh_len_info[time_window=='NWR1',], !is.na(new_ID)) %>% 
  # ggplot(aes(new_ID, value)) +
  geom_violin(aes(variable, value, colour=variable, fill=variable), alpha=.3, lwd=.3) +
  geom_hline(yintercept=5e-5, color="grey20", linetype="dashed", size=.4) +
  theme_bw() +
  scale_y_continuous(limits=c(0, 1e-3)) +
  theme_bw() + 
  theme(legend.position = "none", axis.title=element_text(size=5), axis.text.x=element_text(size=4, angle=20, hjust=1), axis.text.y=element_text(size=5)) +
  labs(x=NULL, y="Heterozygosity per 1Mbp window")

ggplot(SWR1, aes(variable, value, colour=variable)) + 
  # filter(roh_len_info[time_window=='NWR1',], !is.na(new_ID)) %>% 
  # ggplot(aes(new_ID, value)) +
  geom_violin(aes(variable, value, colour=variable, fill=variable), alpha=.3, lwd=.3) +
   geom_hline(yintercept=5e-5, color="grey20", linetype="dashed", size=.4) +
  theme_bw() +
  scale_y_continuous(limits=c(0, 1e-3)) +
  theme_bw() + 
  theme(legend.position = "none", axis.title=element_text(size=5), axis.text.x=element_text(size=4, angle=20, hjust=1), axis.text.y=element_text(size=5)) +
  labs(x=NULL, y="Heterozygosity per 1Mbp window")

ggplot(SWR2, aes(variable, value, colour=variable)) + 
  # filter(roh_len_info[time_window=='NWR1',], !is.na(new_ID)) %>% 
  # ggplot(aes(new_ID, value)) +
  geom_violin(aes(variable, value, colour=variable, fill=variable), alpha=.3, lwd=.3) +
  geom_hline(yintercept=5e-5, color="grey20", linetype="dashed", size=.4) +
  theme_bw() +
  scale_y_continuous(limits=c(0, 1e-3)) + 
  theme_bw() + 
  theme(legend.position = "none", axis.title=element_text(size=5), axis.text.x=element_text(size=4, angle=20, hjust=1), axis.text.y=element_text(size=5)) +
  labs(x=NULL, y="Heterozygosity per 1Mbp window")


```


Based on these plots we chose 5e-5 as the threshold of low heterozygosity to flag a long-enough region as a RoH. Also we excluded after this we ran the upcoming tests for both the whole sample set and for a set without 3 outliers driven by missing data due to depth of coverage: CD-un.1, CD1911.2, ZA1845.1




## 5. RoH length and FRoH


### 5.a) RoH length and FRoH across entire dataset

Read in data on RoH number and length for each sample and reshape appropriately
```{r}

# read in the file to determine the max number of columns
x <- scan("WR_RoH_perscaffold.txt", what = "", sep = "\n")
x <- strsplit(x, "[ \t]+") # split string by white space
max.col <- max(sapply(x, length))

# specify col.names as ?read.table suggests and set old header to missing values
cn <- paste("RoH", 1:max.col, sep = "")
roh <- read.table("WR_RoH_perscaffold.txt", fill = TRUE, col.names = cn, na.strings = "RoH")

# delete old header (first row)
roh = roh[-1,]

```


Calculate length of each RoH per sample
```{r}
# calculate length of RoH per sample in bp
roh_len = roh*1000000 - ((roh-1)*500000)
```


Add relevant metadata from the 'data' dataframe
```{r}
# bind relevant metadata to the RoHs length
new_ID = data$new_ID
country = data$country
time_window = data$time_window

roh_len_info = cbind(new_ID, country, time_window, roh_len)

# melt data with metadata variables as ID
roh_len_info = reshape2::melt(roh_len_info, id.vars = c("new_ID", "country", "time_window"))
```



Prepare a data frame like roh_len_info without the outliers
```{r}
# filtering 3 outlier samples out
roh_len_info_filt = roh_len_info[roh_len_info$new_ID != 'CD-un.1' & roh_len_info$new_ID != 'CD1911.2' & roh_len_info$new_ID != 'ZA1845.1', ]
```


Filter out isolated windows of low heterozygosity (not considered RoH)
```{r}
roh_len_info_above1Mb = roh_len_info[roh_len_info$value > 1000000,]
roh_len_info_filt_above1Mb = roh_len_info_filt[roh_len_info_filt$value > 1000000,]
```


Calculate summary statistics for RoH length to include in plots for both the entire sample set and without the 3 outliers
```{r}
# stats for all samples
group = roh_len_info_above1Mb %>% filter(!is.na(time_window)) %>% 
  group_by(time_window) %>%
  summarize(min=round((min(value)/1000000), digits=2), max=round((max(value)/1000000), digits=2), 
            mean=round((mean(value)/1000000), digits=2))

# stats for filtered samples (-3 outliers)
group_filt = roh_len_info_filt_above1Mb %>% filter(!is.na(time_window)) %>% 
  group_by(time_window) %>%
  summarize(min=round((min(value)/1000000), digits=2), max=round((max(value)/1000000), digits=2), 
            mean=round((mean(value)/1000000), digits=2))

```


### Distribution of RoH length per group

Visualize the distribution of RoH length per group for the total dataset (minus outliers)
```{r}
# remove extra NA category mysteriously created by R
roh_len_all = filter(roh_len_info_filt_above1Mb, !is.na(time_window)) %>% 
  
  # plot violins
  ggplot(aes(time_window, value)) +
  theme_bw() + theme(legend.position = "none", axis.text=element_text(size=7, colour='gray10'), axis.title.y=element_text(size=7), 
                     panel.grid.major=(element_line(size=.12, colour='gray80'))) +
  
geom_violin(aes(time_window, value, colour=time_window, fill=time_window), alpha=.3, lwd=.4) +

  # add colours and filling colours
  scale_fill_manual(values= (c("gold3", "darkorange", "royalblue4", "steelblue3"))) + 
  scale_color_manual(values= (c("gold3", "darkorange", "royalblue4", "steelblue3"))) +
  
  # add stats
  stat_summary(fun=mean, size=0.14, fun.min = function (value) mean(value) - sd(value), fun.max = function(value) mean(value) + sd(value), geom = "pointrange", stroke=.5) +
  
  # embellish
  scale_y_continuous(limits=c(0, 42e6), breaks=c(1.5e6, 10e6, 20e6, 30e6, 40e6), labels=c(1.5, 10, 20, 30, 40)) +
  
  labs(x=NULL, y="RoH length (Mbp)")

roh_len_all
```



Visualize the distribution of RoH length per group for the subsampled dataset (minus outliers)
```{r}
# remove extra NA category mysteriously created by R
roh_len_ss = filter(roh_len_info_filt_above1Mb_ss, !is.na(time_window)) %>% 
  
  # plot violins
  ggplot(aes(time_window, value)) +

  theme_bw() + theme(legend.position = "none", axis.text=element_text(size=7, colour='gray10'), axis.title.y=element_text(size=7), 
                     panel.grid.major=(element_line(size=.12, colour='gray80'))) +
  
  geom_violin(aes(time_window, value, colour=time_window, fill=time_window), alpha=.3, lwd=.4) +

  # add colours and filling colours
  scale_fill_manual(values= (c("gold3", "darkorange", "royalblue4", "steelblue3"))) + 
  scale_color_manual(values= (c("gold3", "darkorange", "royalblue4", "steelblue3"))) +
  
  # add stats
  stat_summary(fun=mean, size=0.14, fun.min = function (value) mean(value) - sd(value), fun.max = function(value) mean(value) + sd(value), geom = "pointrange", stroke=.5) +
  
  
  # embellish
  scale_y_continuous(limits=c(0, 42e6), breaks=c(1.5e6, 10e6, 20e6, 30e6, 40e6), labels=c(1.5, 10, 20, 30, 40)) +
  labs(x=NULL, y="RoH length (Mbp)")


roh_len_ss

```





#### Calculating Froh

Calculate FRoH as an estimate of inbreeding by dividing total length falling in RoH by the total genomic length scanned for each sample
```{r}
# transpose and assign sample IDs as column names
roh_f = t(roh_len)
colnames(roh_f) = data$new_ID

# turn values equal or lower to 1Mbp into missing data
roh_f[roh_f<=1000000]=NA

# sum all remaning values per column and divide by total length of windows screened
# the total length screened is the number of windows multiplied by 1Mbp, minus 
# number of windows minus 1 multiplied by 0.5 Mbp ->
# -> (3659 windows*1Mbp) - ((3659-1)*0.5Mbp)

Froh = as.data.frame(colSums(roh_f, na.rm=T) / 1830000000)
colnames(Froh) = "froh"

# add metadata
new_ID = data$new_ID
country = data$country
time_window = data$time_window

# create dataframes for all samples and for filtered set
Froh_f = cbind(new_ID, country, time_window, Froh)
Froh_f_filt = Froh_f[Froh_f$new_ID != 'CD-un.1' & Froh_f$new_ID != 'CD1911.2' & Froh_f$new_ID != 'ZA1845.1', ]

```


Get summary statistics for Froh across groups for both the unfiltered and filtered sample sets
```{r}
Froh_f %>% 
  group_by(time_window) %>%
  summarize(min=min(froh, na.rm = TRUE), max=max(froh, na.rm = TRUE), median = median(froh, na.rm = TRUE))

Froh_f_filt %>%
  group_by(time_window) %>%
  summarize(min=min(froh, na.rm = TRUE), max=max(froh, na.rm = TRUE), median = median(froh, na.rm = TRUE))
```


Statistical comparison of Froh median between pre- and post-bottleneck samples for each subspecies for the unfiltered data (reported only in supplementary material)
```{r}
# data subsetting
nwr1 = Froh_f[Froh_f$time_window == "NWR1",]
nwr2 = Froh_f[Froh_f$time_window == "NWR2",]
swr1 = Froh_f[Froh_f$time_window == "SWR1",]
swr2 = Froh_f[Froh_f$time_window == "SWR2",]

# calculating medians
median(nwr1$froh)
median(nwr2$froh)
median(swr1$froh)
median(swr2$froh)

# comparing medians of Froh between pre- and post-bottleneck
wilcox.test(nwr1$froh, nwr2$froh, paired = F)
wilcox.test(swr1$froh, swr2$froh, paired = F)
```


Statistical comparison of Froh median between pre- and post-bottleneck samples for each subspecies for the filtered data
```{r}
# data subsetting
nwr1 = Froh_f_filt[Froh_f_filt$time_window == "NWR1",]
nwr2 = Froh_f_filt[Froh_f_filt$time_window == "NWR2",]
swr1 = Froh_f_filt[Froh_f_filt$time_window == "SWR1",]
swr2 = Froh_f_filt[Froh_f_filt$time_window == "SWR2",]

# calculating medians
median(nwr1$froh)
median(nwr2$froh)
median(swr1$froh)
median(swr2$froh)

# comparing medians of Froh between pre- and post-bottleneck
wilcox.test(nwr1$froh, nwr2$froh, paired = F)
wilcox.test(swr1$froh, swr2$froh, paired = F)
```



#### Visualizing Froh

```{r}
fmt_dcimals <- function(decimals=0){
    function(x) format(x,nsmall = decimals,scientific = FALSE)
}
```

Plot Froh across groups after removing the 3 outliers
```{r}
###
# plot violins per group with stats for filtered sample set but not subsampled
froh_all = ggplot(Froh_f_filt, aes(time_window, froh)) +
  theme_bw() + theme(legend.position = "none", axis.text=element_text(size=7, colour='gray10'), axis.title.y=element_text(size=7), 
                     panel.grid.major=(element_line(size=.12, colour='gray80')))


froh_all = froh_all +  geom_violin(aes(colour=time_window, fill=time_window), lwd=.3, alpha=.4) +
  
  # add scatter points per group
  geom_jitter(colour='gray40', shape=16, position=position_jitter(0.11), cex=.6, alpha=.7) + 
  
  # add colours and filling colours
  scale_fill_manual(values= (c("gold3", "darkorange", "royalblue4", "steelblue3"))) + 
  scale_color_manual(values= (c("gold3", "darkorange", "royalblue4", "steelblue3"))) +
  
  # add stats and an unpaired wilcox test to compare the medians
  stat_summary(fun=median, geom="crossbar", colour='black', width=.12, lwd=.3) +
  
  stat_summary(fun=mean, fun.min = function (corrected_gw_het) mean(corrected_gw_het) - sd(corrected_gw_het), 
               fun.max = function(corrected_gw_het) mean(corrected_gw_het) + sd(corrected_gw_het), geom = "pointrange" , size=.12, stroke=.4, col="black") +
  
  stat_compare_means(paired=F, method = "wilcox.test", comparisons= list(c("NWR1", "NWR2"), c("SWR1", "SWR2")), label.y=c(0.4, 0.5), size=2.5) +
  
  # embellish
  scale_y_continuous(limits=c(0.1, 0.53),labels = function(x) format(x, scientific = FALSE)) +
  
  labs(x=NULL, y="FRoH")

froh_all
```


##### Make RoH len and Froh figure for main text (Figure 4 in WR v3)
```{r}
pdf('WR_v3_Fig4_v1.pdf', height=4, width=5, useDingbats = F)
plot_grid(roh_len_all, froh_all, nrow=1, ncol=2, labels=c('A','B'), label_size=7)
dev.off()
```



### 5.b) RoH length and FRoH in subsampled dataset


Two out of the three outlier samples removed in RoH analyses, orginate from DRC, so they would be exluded anyway from the calculations below
That would leave only one out of three outliers in the subsampled dataset, so we chose to perform these calculations only on the filtered dataset, without any of the 3 outliers
Subsampling by country is therefore an added filtering step on top of removing the 3 outliers (n = 37 final number of samples)

Take already parsed data for RoH length after removing outliers and filter out pertinent samples (from DRC and captive) 
```{r}
roh_len_info_filt_above1Mb_ss = subset(roh_len_info_filt_above1Mb, country != 'DRC' & country != 'captive' | is.na(country))
```


Calculate summary statistics for RoH length to include in plots for both the entire sample set and without the 3 outliers
```{r}
group_filt_ss = roh_len_info_filt_above1Mb_ss %>% filter(!is.na(time_window)) %>% 
  group_by(time_window) %>%
  summarize(min=round((min(value)/1000000), digits=2), max=round((max(value)/1000000), digits=2), 
            mean=round((mean(value)/1000000), digits=2))

```


Take the relevant values of FRoH, those corresponding to the samples present in the subsampled dataset
```{r}
Froh_f_filt_ss = subset(Froh_f_filt, country != 'DRC' & country != 'captive' | is.na(country))
```


Get summary statistics for FRoH across groups for the subsampled dataset without outliers (n = 37)
```{r}
Froh_f_filt_ss %>%
  group_by(time_window) %>%
  summarize(min=min(froh, na.rm = TRUE), max=max(froh, na.rm = TRUE), median = median(froh, na.rm = TRUE))
```


Statistical comparison of Froh median between pre- and post-bottleneck samples for each subspecies for the subsampled data (n = 37)
```{r}
# data subsetting
nwr1 = Froh_f_filt_ss[Froh_f_filt_ss$time_window == "NWR1",]
nwr2 = Froh_f_filt_ss[Froh_f_filt_ss$time_window == "NWR2",]
swr1 = Froh_f_filt_ss[Froh_f_filt_ss$time_window == "SWR1",]
swr2 = Froh_f_filt_ss[Froh_f_filt_ss$time_window == "SWR2",]

# comparing medians of Froh between pre- and post-bottleneck
wilcox.test(nwr1$froh, nwr2$froh, paired = F)
wilcox.test(swr1$froh, swr2$froh, paired = F)
```


Visualize Froh across groups
```{r}
###
# plot violins per group with stats for filtered sample set but not subsampled
froh_ss = ggplot(Froh_f_filt_ss, aes(time_window, froh)) +
  theme_bw() + theme(legend.position = "none", axis.text=element_text(size=7, colour='gray10'), axis.title.y=element_text(size=7), 
                     panel.grid.major=(element_line(size=.12, colour='gray80')))


froh_ss = froh_ss +  geom_violin(aes(colour=time_window, fill=time_window), lwd=.3, alpha=.4) +
  
  # add scatter points per group
  geom_jitter(colour='gray40', shape=16, position=position_jitter(0.11), cex=.6, alpha=.7) + 
  
  # add colours and filling colours
  scale_fill_manual(values= (c("gold3", "darkorange", "royalblue4", "steelblue3"))) + 
  scale_color_manual(values= (c("gold3", "darkorange", "royalblue4", "steelblue3"))) +
  
  # add stats and an unpaired wilcox test to compare the medians
  stat_summary(fun=median, geom="crossbar", colour='black', width=.12, lwd=.3) +
  
  stat_summary(fun=mean, fun.min = function (corrected_gw_het) mean(corrected_gw_het) - sd(corrected_gw_het), 
               fun.max = function(corrected_gw_het) mean(corrected_gw_het) + sd(corrected_gw_het), geom = "pointrange" , size=.12, stroke=.4, col="black") +
  
  stat_compare_means(paired=F, method = "wilcox.test", comparisons= list(c("NWR1", "NWR2"), c("SWR1", "SWR2")), label.y=c(0.4, 0.5), size=2.5) +
  
  # embellish
  scale_y_continuous(limits=c(0.1, 0.53),labels = function(x) format(x, scientific = FALSE)) +
  
  labs(x=NULL, y="FRoH")

froh_ss
```




##### Make RoH len and Froh figure for supplementary (Figure S9 in WR v3)
```{r}
pdf('WR_v3_FigS9_v1.pdf', height=4, width=5, useDingbats = F)
plot_grid(roh_len_ss, froh_ss, nrow=1, ncol=2, labels=c('A','B'), label_size=7)
dev.off()
```



#########################################################################################################

# Figures in WR v2

## Making Figure on genomic erosion for main text
Combine plots for Figure 2 in one page
```{r eval=FALSE, include=FALSE}
pdf('WR_v2_Fig2_v1.pdf', height=4.5, width=5, useDingbats=F)
plot_grid(pi, gw_het, exon_het, froh_plot, labels=c('A','B','C','D'), label_size=7, nrow=2 , ncol=2)
dev.off()
```


## Making Figure S4
Combine plots for Figure S4 in one page
```{r eval=FALSE, include=FALSE}
pdf('WR_v2_FigS4_v1.pdf', height=3.5, useDingbats=F)
plot_grid(gw_het_ss, exon_het_ss, froh_plot_ss, labels=c('A','B','C'), label_size=9, nrow=1 , ncol=3)
dev.off()
```

## Making Figure S3 in WR v2 or S7 in WR v3
### Distribution of local estimates of heterozygosity per sample

Visualize the distribution of local estimates of heterozygosity per sample across windows in order to determine the threshold of heterozygosity below which long stretches will be flagged as RoH
```{r}
A = ggplot(NWR1, aes(variable, value, colour=variable)) + 
  # filter(roh_len_info[time_window=='NWR1',], !is.na(new_ID)) %>% 
  # ggplot(aes(new_ID, value)) +
  geom_violin(aes(variable, value, colour=variable, fill=variable), alpha=.3, lwd=.3) +
  theme_bw() +
  scale_y_continuous(limits=c(0, 3e-3)) +
  geom_hline(yintercept=5e-5, color="grey20", linetype="dashed", size=.4) +
  theme_bw() + 
  theme(legend.position = "none", axis.title=element_text(size=5), axis.text.x=element_text(size=4, angle=20, hjust=1, colour='gray10'), axis.text.y=element_text(size=5, colour='gray10')) +
  labs(x=NULL, y="Heterozygosity per 1Mbp window")

B = ggplot(NWR2, aes(variable, value, colour=variable)) + 
  # filter(roh_len_info[time_window=='NWR1',], !is.na(new_ID)) %>% 
  # ggplot(aes(new_ID, value)) +
  geom_violin(aes(variable, value, colour=variable, fill=variable), alpha=.3, lwd=.3) +
  geom_hline(yintercept=5e-5, color="grey20", linetype="dashed", size=.4) +
  theme_bw() +
  scale_y_continuous(limits=c(0, 3e-3)) +
  theme_bw() + 
  theme(legend.position = "none", axis.title=element_text(size=5), axis.text.x=element_text(size=4, angle=20, hjust=1, colour='gray10'), axis.text.y=element_text(size=5, colour='gray10')) +
  labs(x=NULL, y="Heterozygosity per 1Mbp window")

C = ggplot(SWR1, aes(variable, value, colour=variable)) + 
  # filter(roh_len_info[time_window=='NWR1',], !is.na(new_ID)) %>% 
  # ggplot(aes(new_ID, value)) +
  geom_violin(aes(variable, value, colour=variable, fill=variable), alpha=.3, lwd=.3) +
   geom_hline(yintercept=5e-5, color="grey20", linetype="dashed", size=.4) +
  theme_bw() +
  scale_y_continuous(limits=c(0, 3e-3)) +
  theme_bw() + 
  theme(legend.position = "none", axis.title=element_text(size=5), axis.text.x=element_text(size=4, angle=20, hjust=1, colour='gray10'), axis.text.y=element_text(size=5, colour='gray10')) +
  labs(x=NULL, y="Heterozygosity per 1Mbp window")

D = ggplot(SWR2, aes(variable, value, colour=variable)) + 
  # filter(roh_len_info[time_window=='NWR1',], !is.na(new_ID)) %>% 
  # ggplot(aes(new_ID, value)) +
  geom_violin(aes(variable, value, colour=variable, fill=variable), alpha=.3, lwd=.3) +
  geom_hline(yintercept=5e-5, color="grey20", linetype="dashed", size=.4) +
  theme_bw() +
  scale_y_continuous(limits=c(0, 3e-3)) + 
  theme_bw() + 
  theme(legend.position = "none", axis.title=element_text(size=5), axis.text.x=element_text(size=4, angle=20, hjust=1, colour='gray10'), axis.text.y=element_text(size=5, colour='gray10')) +
  labs(x=NULL, y="Heterozygosity per 1Mbp window")


```

##### Save individual distribution of local het as Figure S7 in WR v3
```{r}
pdf('WR_v3_FigS7_v1.pdf', height=5, width=6, useDingbats = F)
plot_grid(A, B, C, D, labels=c("A", "B", "C", "D"), label_size=7, ncol = 2, nrow = 2)
dev.off()
```


### Distribution of RoH length per group

Visualize the distribution of RoH length per group for both the total dataset (minus outliers) and the subsampled dataset (minus outliers)
```{r include=FALSE}
# remove extra NA category mysteriously created by R
roh_len_all = filter(roh_len_info_filt_above1Mb, !is.na(time_window)) %>% 
  
  # plot violins
  ggplot(aes(time_window, value)) +
  geom_violin(mapping=aes(time_window, value, colour=time_window, fill=time_window), alpha=.3, lwd=.3) +

  # add colours and filling colours
  scale_fill_manual(values= (c("gold2", "darkorange", "royalblue4", "steelblue3"))) + 
  scale_color_manual(values= (c("gold2", "darkorange", "royalblue4", "steelblue3"))) +
  
  # add stats
  stat_summary(fun.y=mean, size=0.14, fun.ymin = function (value) mean(value) - sd(value), fun.ymax = function(value) mean(value) + sd(value), geom = "pointrange", stroke=.5) +
  
  # embellish
  scale_y_continuous(limits=c(0, 42e6), breaks=c(1.5e6, 10e6, 20e6, 30e6, 40e6), labels=c(1.5, 10, 20, 30, 40)) +
  theme_bw() + theme(legend.position = "none", axis.title=element_text(size=5), axis.text=element_text(size=5)) +
  labs(x=NULL, y="RoH length (Mbp)")


# remove extra NA category mysteriously created by R
roh_len_ss = filter(roh_len_info_filt_above1Mb_ss, !is.na(time_window)) %>% 
  
  # plot violins
  ggplot(aes(time_window, value)) +
  geom_violin(mapping=aes(time_window, value, colour=time_window, fill=time_window), alpha=.3, lwd=.3) +

  # add colours and filling colours
  scale_fill_manual(values= (c("gold2", "darkorange", "royalblue4", "steelblue3"))) + 
  scale_color_manual(values= (c("gold2", "darkorange", "royalblue4", "steelblue3"))) +
  
  # add stats
  stat_summary(fun.y=mean, size=0.14, fun.ymin = function (value) mean(value) - sd(value), fun.ymax = function(value) mean(value) + sd(value), geom = "pointrange", stroke=.5) +
  
  # embellish
  scale_y_continuous(limits=c(0, 42e6), breaks=c(1.5e6, 10e6, 20e6, 30e6, 40e6), labels=c(1.5, 10, 20, 30, 40)) +
  theme_bw() + theme(legend.position = "none", axis.title=element_text(size=5), axis.text=element_text(size=5)) +
  labs(x=NULL, y="RoH length (Mbp)")




```

Save
```{r}
pdf('WR_v2_FigS3_v1.pdf', height=7, useDingbats = F)
plot_grid(A, B, C, D, roh_len_all, roh_len_ss, labels=c("A", "B", "C", "D", 'E', 'F'), label_size=7, ncol = 2, nrow = 3)
dev.off()
```


EXTRA STUFF

Visualize the distribution of RoH length (including isolated windows of 1Mbp) per sample
```{r include=FALSE}

filter(roh_len_info[time_window=='NWR1',], !is.na(new_ID)) %>% 
  ggplot(aes(new_ID, value)) +
  geom_violin(mapping=aes(new_ID, value, colour=new_ID, fill=new_ID), alpha=.3, lwd=.3) +
  theme_bw() +
  scale_y_continuous(limits=c(0, 41e6), breaks=c(1.5e6, 10e6, 20e6, 30e6, 40e6), labels=c(1.5, 10, 20, 30, 40)) +
  theme_bw() + 
  theme(legend.position = "none", axis.title=element_text(size=5), axis.text.x=element_text(size=4, angle=20, hjust=1), axis.text.y=element_text(size=5)) +
  labs(x=NULL, y="Length of low-heterozygosity stretches (Mbp)")

filter(roh_len_info[time_window=='NWR2',], !is.na(new_ID)) %>% 
  ggplot(aes(new_ID, value)) +
  geom_violin(mapping=aes(new_ID, value, colour=new_ID, fill=new_ID), alpha=.3, lwd=.3) +
  theme_bw() +
  scale_y_continuous(limits=c(0, 41e6), breaks=c(1.5e6, 10e6, 20e6, 30e6, 40e6), labels=c(1.5, 10, 20, 30, 40)) +
  theme_bw() + 
  theme(legend.position = "none", axis.title=element_text(size=5), axis.text.x=element_text(size=4, angle=20, hjust=1), axis.text.y=element_text(size=5)) +
  labs(x=NULL, y="Length of low-heterozygosity stretches (Mbp)")

filter(roh_len_info[time_window=='SWR1',], !is.na(new_ID)) %>% 
  ggplot(aes(new_ID, value)) +
  geom_violin(mapping=aes(new_ID, value, colour=new_ID, fill=new_ID), alpha=.3, lwd=.3) +
  theme_bw() +
  scale_y_continuous(limits=c(0, 41e6), breaks=c(1.5e6, 10e6, 20e6, 30e6, 40e6), labels=c(1.5, 10, 20, 30, 40)) +
  theme_bw() + 
  theme(legend.position = "none", axis.title=element_text(size=5), axis.text.x=element_text(size=4, angle=20, hjust=1), axis.text.y=element_text(size=5)) +
  labs(x=NULL, y="Length of low-heterozygosity stretches (Mbp)")

filter(roh_len_info[time_window=='SWR2',], !is.na(new_ID)) %>% 
  ggplot(aes(new_ID, value)) +
  geom_violin(mapping=aes(new_ID, value, colour=new_ID, fill=new_ID), alpha=.3, lwd=.3) +
  theme_bw() +
  scale_y_continuous(limits=c(0, 41e6), breaks=c(1.5e6, 10e6, 20e6, 30e6, 40e6), labels=c(1.5, 10, 20, 30, 40)) +
  theme_bw() + 
  theme(legend.position = "none", axis.title=element_text(size=5), axis.text.x=element_text(size=4, angle=20, hjust=1), axis.text.y=element_text(size=5)) +
  labs(x=NULL, y="Length of low-heterozygosity stretches (Mbp)")
```




