---
title: "WR genomic diversity (pi and het)"
author: "FÃ¡tima S. Barreiro"
date: "April 21, 2020"
output: html_document
---

# Analyses and visualization of group, and individual-level metrics of genomic diversity


Load required libraries
```{r message=FALSE, warning=FALSE, paged.print=FALSE}
library(ggplot2)
library(cowplot)
library(ggrepel)
library(reshape2)
library(dplyr)
library(tidyverse)
library(data.table)
require(scales)
library(ggpubr)
library(tidyr)
library(PairedData)
library(gridExtra)
```




## 1. Nucleotide diversity based on transversion sites per group 

Read in data
```{r include=FALSE}
pi_NWR = read.csv("NWR_nucDiversity_tv.csv", header=T)
pi_SWR = read.csv("SWR_nucDiversity_tv.csv", header=T)
```

Bind relevant columns and reshape
```{r}
pi_WR = as.data.frame(cbind(pi_NWR$pi_NWR1, pi_NWR$pi_NWR2, pi_SWR$pi_SWR1, pi_SWR$pi_SWR2))
colnames(pi_WR) = c("NWR1", "NWR2", "SWR1", "SWR2")

pi_WR = reshape2::melt(pi_WR)

```

Summary statistics for pi_tv across the 4 groups
```{r}
pi_WR %>%
  group_by(variable) %>%
  summarize(min=min(value), max=max(value), median=median(value))
```


Statistical comparisons of medians between pi-tv in pre- and post-bottleneck samples per subspecies
```{r}
pi_nwr1 = subset(pi_WR,  variable == "NWR1", value, drop = TRUE)
pi_nwr2 = subset(pi_WR,  variable == "NWR2", value, drop = TRUE)
pi_swr1 = subset(pi_WR,  variable == "SWR1", value, drop = TRUE)
pi_swr2 = subset(pi_WR,  variable == "SWR2", value, drop = TRUE)


wilcox.test(pi_nwr1, pi_nwr2, paired=TRUE)
wilcox.test(pi_swr1, pi_swr2, paired=TRUE)
```


Visualization
```{r}
# plot
pi = ggplot(pi_WR, aes(variable, value, fill=variable))  +
   theme_bw() + theme(legend.position = "none", axis.text=element_text(size=7, colour='gray10'), axis.title.y=element_text(size=7), 
                     panel.grid.major=(element_line(size=.12, colour='gray80')))
  
  # format scatter points
pi = pi +
    # format boxplots
  geom_boxplot(alpha=.2, lwd=.2, fill='gray30') +
  scale_color_manual(values= (c("gold3", "darkorange", "royalblue4", "steelblue3"))) +
  # add the points
  geom_jitter(aes(colour=variable), shape=17, position=position_jitter(0.11), cex=.7, alpha=.7) +  
  
  # get summary stats and groups comparison
  stat_compare_means(paired=TRUE, method = "wilcox.test", comparisons= list(c("NWR1", "NWR2")), label="p.format", label.y=0.00064, size=2.5) +
  stat_compare_means(paired=TRUE, method = "wilcox.test", comparisons= list(c("SWR1", "SWR2")), label="p.format", label.y=0.00044, size=2.5) +
  
  # embellish
  scale_y_continuous(limits=c(0.0001,0.00068), labels = function(x) format(x, scientific = FALSE)) +
  labs(x=NULL, y="Nucleotide diversity (tv)") + 
  theme_bw() + theme(legend.position = "none", axis.text=element_text(size=7, colour='gray10'), axis.title.y=element_text(size=7), 
                     panel.grid.major=(element_line(size=.12, colour='gray80'))) 
  
pi
```

####################################################
Missing: Pi-tv for subsampled dataset (running in computerome)
Did not work* 
################################################






## 2. Individual measures of genomic diversity: Statistical comparisons between time windows and delta estimators 


Reading in file containing metadata and genomic diversity metrics 
```{r}
data = read.csv("WR_v3_TableS1_v1.csv", h=T, fill=T, stringsAsFactors=T, fileEncoding="latin1")
```


Subsetting data per group (subspecies and time window)
```{r}
nwr1 = data[data$time_window == "NWR1",]
nwr2 = data[data$time_window == "NWR2",]
swr1 = data[data$time_window == "SWR1",]
swr2 = data[data$time_window == "SWR2",]
```



### 2.a) Genome-wide heterozygosity

Summary statistics for gw_het across the 4 groups
```{r}
data %>%
  group_by(time_window) %>%
  summarize(min=min(corrected_gw_het), max=max(corrected_gw_het), median=median(corrected_gw_het))
```


Statistical comparison of gw_het median between pre- and post-bottleneck samples for each subspecies
```{r}
wilcox.test(nwr1$corrected_gw_het, nwr2$corrected_gw_het, paired = F)
wilcox.test(swr1$corrected_gw_het, swr2$corrected_gw_het, paired = F)
```


Visualization
```{r}
# plot het values per time window
gw_het = ggplot(data, aes(time_window, corrected_gw_het)) + 
  theme_bw() + theme(legend.position = "none", axis.text=element_text(size=7, colour='gray10'), axis.title.y=element_text(size=7), 
                     panel.grid.major=(element_line(size=.12, colour='gray80')))
  
  
      # plot violin per group
gw_het = gw_het +  geom_violin(aes(colour=time_window, fill=time_window), lwd=.3, alpha=.4) +
  
  # add scatter points per group
  geom_jitter(colour='gray40', shape=16, position=position_jitter(0.11), cex=.6, alpha=.7) + 
  
  # add colours and filling colours
  scale_fill_manual(values= (c("gold3", "darkorange", "royalblue4", "steelblue3"))) + 
  scale_color_manual(values= (c("gold3", "darkorange", "royalblue4", "steelblue3"))) +
  
  # add stats and an unpaired wilcox test to compare the medians
  stat_summary(fun=median, geom="crossbar", colour='black', width=.12, lwd=.3) +
  
  stat_summary(fun=mean, fun.min = function (corrected_gw_het) mean(corrected_gw_het) - sd(corrected_gw_het), 
               fun.max = function(corrected_gw_het) mean(corrected_gw_het) + sd(corrected_gw_het), geom = "pointrange" , size=.12, stroke=.4, col="black") +
  
  stat_compare_means(paired=F, method = "wilcox.test", comparisons= list(c("NWR1", "NWR2"), c("SWR1", "SWR2")), label.y=c(0.00048, 0.00022), size=2.5) +
  
  # embellish
  scale_y_continuous(limits=c(0.00007, 0.0005),labels = function(x) format(x, scientific = FALSE)) +
  
  
  labs(x=NULL, y="Genome-wide heterozygosity (tv)")

gw_het
```



### 2.b) Exon heterozygosity

Summary statistics for exon_het across the 4 groups
```{r}
data %>%
  group_by(time_window) %>%
  summarize(min=min(corrected_exon_het), max=max(corrected_exon_het), median=median(corrected_exon_het))
```


Statistical comparison of exon_het median between pre- and post-bottleneck samples for each subspecies
```{r}
wilcox.test(nwr1$corrected_exon_het, nwr2$corrected_exon_het, paired = F)
wilcox.test(swr1$corrected_exon_het, swr2$corrected_exon_het, paired = F)
```


Visualization
```{r}
exon_het = ggplot(data, aes(time_window, corrected_exon_het, fill=time_window)) + 
  theme_bw() + theme(legend.position = "none", axis.text=element_text(size=7, colour='gray10'), axis.title.y=element_text(size=7), 
                     panel.grid.major=(element_line(size=.12, colour='gray80')))

    # plot violin per group
exon_het = exon_het +  geom_violin(aes(colour=time_window, fill=time_window), lwd=.3, alpha=.4) +
  
  # add scatter points per group
  geom_jitter(colour='gray40', shape=16, position=position_jitter(0.11), cex=.6, alpha=.7) + 
  
  # add colours and filling colours
  scale_fill_manual(values= (c("gold3", "darkorange", "royalblue4", "steelblue3"))) + 
  scale_color_manual(values= (c("gold3", "darkorange", "royalblue4", "steelblue3"))) +
  
  # add stats and an unpaired wilcox test to compare the medians
  stat_summary(fun=median, geom="crossbar", colour='black', width=.12, lwd=.3) +
  
  stat_summary(fun=mean, fun.min = function (corrected_gw_het) mean(corrected_gw_het) - sd(corrected_gw_het), 
               fun.max = function(corrected_gw_het) mean(corrected_gw_het) + sd(corrected_gw_het), geom = "pointrange" , size=.12, stroke=.4, col="black") +
  
  stat_compare_means(paired=F, method = "wilcox.test", comparisons= list(c("NWR1", "NWR2"), c("SWR1", "SWR2")), label.y=c(0.00033, 0.00022), size=2.5) +
  
  # embellish
  scale_y_continuous(limits=c(0.00004, 0.00035),labels = function(x) format(x, scientific = FALSE)) +
  
  
  labs(x=NULL, y="Exon heterozygosity (tv)")

exon_het
  
```


#### Make genomic erosion figure (Figure 3 in WR v3)

First make the panel of heterozygosities
```{r}
HET = plot_grid(gw_het, exon_het, nrow=1, ncol=2, labels=c('B','C'), label_size=7)
HET
```



```{r}
# pdf('WR_v3_Fig3_v1.pdf', height=5, width=5, useDingbats = F)
# plot_grid(pi, HET, nrow=2, ncol=1, labels=c('A', NULL), label_size = 7, rel_heights = c(0.7, 1.2))
# dev.off()
```

```{r}
png('WR_v3_Fig3_v1.png', height=5, width=5, units = 'in', res = 300)
plot_grid(pi, HET, nrow=2, ncol=1, labels=c('A', NULL), label_size = 7, rel_heights = c(0.7, 1.2))
dev.off()
```



#####################################################################################


## 3. Individual diversity metrics in subsampled dataset


Subsampling the full dataset to assess patterns of genomic erosion after mitigated sampling bias
Basically, from NWR1 we exclude samples originating from DRC because that genetic origin is not represented in our set of NWR2
From SWR2 we exclude captive individuals, since they were bred in zoos, so we focus only on wild SWR when comparing pre- and post-bottleneck

```{r}
data_ss = subset(data, country != 'DRC' & country != 'captive' | is.na(country))
```

Subsetting data per group (subspecies and time window)
```{r}
nwr1_ss = data_ss[data_ss$time_window == "NWR1",]
nwr2_ss = data_ss[data_ss$time_window == "NWR2",]
swr1_ss = data_ss[data_ss$time_window == "SWR1",]
swr2_ss = data_ss[data_ss$time_window == "SWR2",]
```



### 3.a) Genome-wide heterozygosity in subsampled dataset


Summary statistics for gw_het across the 4 groups
```{r}
data_ss %>%
  group_by(time_window) %>%
  summarize(min=min(corrected_gw_het), max=max(corrected_gw_het), median=median(corrected_gw_het))
```


Statistical comparison of gw_het median between pre- and post-bottleneck samples for each subspecies
```{r}
wilcox.test(nwr1_ss$corrected_gw_het, nwr2_ss$corrected_gw_het, paired = F)
wilcox.test(swr1_ss$corrected_gw_het, swr2_ss$corrected_gw_het, paired = F)
```


Visualization
```{r}
# plot het values per time window
gw_het_ss = ggplot(data_ss, aes(time_window, corrected_gw_het)) + 
  theme_bw() + theme(legend.position = "none", axis.text=element_text(size=7, colour='gray10'), axis.title.y=element_text(size=7), 
                     panel.grid.major=(element_line(size=.12, colour='gray80')))

    # plot violin per group
gw_het_ss = gw_het_ss +  geom_violin(aes(colour=time_window, fill=time_window), lwd=.3, alpha=.4) +
  
  # add scatter points per group
  geom_jitter(colour='gray40', shape=16, position=position_jitter(0.11), cex=.6, alpha=.7) + 
  
  # add colours and filling colours
  scale_fill_manual(values= (c("gold3", "darkorange", "royalblue4", "steelblue3"))) + 
  scale_color_manual(values= (c("gold3", "darkorange", "royalblue4", "steelblue3"))) +
  
  # add stats and an unpaired wilcox test to compare the medians
  stat_summary(fun=median, geom="crossbar", colour='black', width=.12, lwd=.3) +
  
  stat_summary(fun=mean, fun.min = function (corrected_gw_het) mean(corrected_gw_het) - sd(corrected_gw_het), 
               fun.max = function(corrected_gw_het) mean(corrected_gw_het) + sd(corrected_gw_het), geom = "pointrange" , size=.12, stroke=.4, col="black") +
  
  stat_compare_means(paired=F, method = "wilcox.test", comparisons= list(c("NWR1", "NWR2"), c("SWR1", "SWR2")), label.y=c(0.00023, 0.00022), size=2.5) +
  
  # embellish
  scale_y_continuous(limits=c(0.00008, 0.00025),labels = function(x) format(x, scientific = FALSE)) +
  
  
  labs(x=NULL, y="Genome-wide heterozygosity (tv)")

gw_het_ss
```



### 3.b) Exon heterozygosity in subsampled data


Summary statistics for exon_het across the 4 groups
```{r}
data_ss %>%
  group_by(time_window) %>%
  summarize(min=min(corrected_exon_het), max=max(corrected_exon_het), median=median(corrected_exon_het))
```

Statistical comparison of exon_het median between pre- and post-bottleneck samples for each subspecies
```{r}
wilcox.test(nwr1_ss$corrected_exon_het, nwr2_ss$corrected_exon_het, paired = F)
wilcox.test(swr1_ss$corrected_exon_het, swr2_ss$corrected_exon_het, paired = F)
```

Visualization
```{r}
exon_het_ss = ggplot(data_ss, aes(time_window, corrected_exon_het, fill=time_window)) + 
    theme_bw() + theme(legend.position = "none", axis.text=element_text(size=7, colour='gray10'), axis.title.y=element_text(size=7), 
                     panel.grid.major=(element_line(size=.12, colour='gray80')))

    # plot violin per group
exon_het_ss = exon_het_ss +  geom_violin(aes(colour=time_window, fill=time_window), lwd=.3, alpha=.4) +
  
  # add scatter points per group
  geom_jitter(colour='gray40', shape=16, position=position_jitter(0.11), cex=.6, alpha=.7) + 
  
  # add colours and filling colours
  scale_fill_manual(values= (c("gold3", "darkorange", "royalblue4", "steelblue3"))) + 
  scale_color_manual(values= (c("gold3", "darkorange", "royalblue4", "steelblue3"))) +
  
  # add stats and an unpaired wilcox test to compare the medians
  stat_summary(fun=median, geom="crossbar", colour='black', width=.12, lwd=.3) +
  
  stat_summary(fun=mean, fun.min = function (corrected_gw_het) mean(corrected_gw_het) - sd(corrected_gw_het), 
               fun.max = function(corrected_gw_het) mean(corrected_gw_het) + sd(corrected_gw_het), geom = "pointrange" , size=.12, stroke=.4, col="black") +
  
  stat_compare_means(paired=F, method = "wilcox.test", comparisons= list(c("NWR1", "NWR2"), c("SWR1", "SWR2")), label.y=c(0.00014, 0.00014), size=2.5) +
  
  # embellish
  scale_y_continuous(limits=c(0.00005, 0.00015),labels = function(x) format(x, scientific = FALSE)) +
  
  
  labs(x=NULL, y="Exon heterozygosity (tv)")

exon_het_ss
```



#### Make Figure on individual genomic erosion for supplementary 
```{r}
# pdf('WR_v3_FigS8.pdf', height=4, width=5, useDingbats = F)
# plot_grid(gw_het_ss, exon_het_ss, nrow=1, ncol=2, labels=c('A','B'), label_size=7)
# dev.off()
```

```{r}
png('WR_v3_FigS8_v2.png', height=4, width=5, units = 'in', res = 500)
plot_grid(gw_het_ss, exon_het_ss, nrow=1, ncol=2, labels=c('A','B'), label_size=7)
dev.off()
```



