---
title: "WR_v4_geneticLoad"
author: "FÃ¡tima S. Barreiro"
date: "November 13, 2020"
output: html_document
---



## Build a database that includes the black rhino ref and annotation that was used for mapping
```{bash}
# in the snpEff folder:
## copy the br ref used to the snpEff/data/genomes/ folder under the name black_rhino.fa
## copy the corresponding gff file to snpEff/data/black_rhino folder (newly made) under the number of genes.gff
## modify the snpEff.config file to include the black rhino specific lines
### the black rhino lines are:
nano WRtoBR_lines.txt
        # BlackRhino, version 19Dec2016_S9zPn
        black_rhino.genome : BlackRhino

### take template config file, and add the black rhino lines
cat temp.config2 WRtoBR_lines.txt > snpEff.config3

### change the name of the modified config file
cp snpEff.config3 snpEff.config

### build the database with snpEff
java -jar snpEff.jar build -gff3 -v black_rhino

```


## Call genotypes for each sample separately with the bam files from mapping these white rhinos against the black rhino assembly
```{bash}

## generate all sites vcfs
for file in $(cat SWR_bamlist.txt); do BN=$(echo $file | awk -F/ '{print $NF}'| cut -d '.' -f1); java -Djava.io.tmpdir=/home/scratch/ -XX:ParallelGCThreads=16 -XX:+UseParallelGC -XX:-UsePerfData -Xms5g -Xmx60g -jar /GenomeAnalysisTK/GenomeAnalysisTK.jar -T HaplotypeCaller -R /PATH/TO/BLACK/RHINO/REF/ASSEMBLY/black_rhino_ref.fasta --output_mode EMIT_ALL_SITES -I $file -o $BN".allsites.vcf"  -nct 5 -ERC BP_RESOLUTION --min_base_quality_score 20 -mmq 30 ; done

## compress as produced with bgzip
for file in *vcf ; do bgzip $file ; done 

## make a tabix index for each created vcf 
module load tabix/1.2.1 
for file in *vcf.gz ; do tabix -p vcf $file ; done 

## Convert to gvcf files
for file in *.allsites.vcf.gz ; do basename=$(echo $file | cut -d '.' -f1) ; java -Djava.io.tmpdir=/home/people/fasaba/scratch/ -jar /GenomeAnalysisTK/GenomeAnalysisTK.jar -T GenotypeGVCFs -R /PATH/TO/BLACK/RHINO/REF/ASSEMBLY/black_rhino_ref.fasta --variant $file --includeNonVariantSites -o "$basename".g.vcf" ; done 

## compress as produced with bgzip
for file in *g.vcf ; do bgzip $file ; done 

## make a tabix index for each created vcf 
module load tabix/1.2.1 
for file in *g.vcf.gz ; do tabix -p vcf $file ; done 

```


## Filter sites from genomic vcfs and annotate homozygous derived sites of at least 10x with snpEff 
```{bash}
# snpEff annotations for only:
# biallelic sites
# homozygous derived
# min depth of 10x

# filter and annotate the sites 
for vcf in /PATH/TO/VCF/FILES/*.g.vcf.gz ; do name=$(echo $vcf | awk -F/ '{print $NF}' | sed 's/.g.vcf.gz//g') ; zcat $vcf | bcftools view -c 1 -g hom -i 'TYPE="snp" && FORMAT/DP>9' $vcf | java -Xmx4g -jar ./snpEff/snpEff.jar -c ./snpEff/snpEff.config -v black_rhino | bgzip > $name.Black_rhino.snpeff.minDP10.homD.vcf.gz ; done

# parse output with snpSift
for file in *.snpeff.minDP10.homD.vcf.gz ; do sample=$(echo $file | cut -d'.' -f1-5); echo $sample ; zcat $file | ./snpEff/scripts/vcfEffOnePerLine.pl | java -jar ./snpEff/SnpSift.jar extractFields - CHROM POS REF ALT "EFF[*].IMPACT" "EFF[*].GENE" "EFF[*].EFFECT" > $sample.EffList ; done
```



## To estimate genetic load, use script LoadStats.sh that takes as parameters one input file: an effect list of a sample
```{bash}
#!/bin/bash

sample=$(echo $1 | cut -d'.' -f1,4,5)

rm $sample.LoadStats
touch $sample.LoadStats



# number of synonymous variants
echo "synonymous" >> $sample.LoadStats
zcat $1 | grep synonymous | wc -l >> $sample.LoadStats

# number of missense variants 
echo "missense" >> $sample.LoadStats
zcat $1 | grep missense | wc -l >> $sample.LoadStats

# total number of LOF variants 
echo "total_LoF" >> $sample.LoadStats
zcat $1 | grep HIGH | grep -e stop_gained -e frameshift_variant -e splice_acceptor_variant -e splice_donor_variant | wc -l >> $sample.LoadStats

# number of genes harbouring at least one LOF variant
echo "genes_min1_LoF" >> $sample.LoadStats
zcat $1 | grep HIGH | grep -e stop_gained -e frameshift_variant -e splice_acceptor_variant -e splice_donor_variant | cut -f6 | sort | uniq | wc -l >> $sample.LoadStats

# save list of unique genes with at least one LOF
#zcat $1 | grep -v intergenic | grep HIGH | grep -e stop_gained -e frameshift_variant -e splice_acceptor_variant -e splice_donor_variant | cut -f6 | sort | uniq > $sample.minDP8.genesLOF


# parse the output
cat $sample.LoadStats | paste - - | awk '{for (i=1; i<=NF; i++) a[i,NR]=$i ; max=(max<NF?NF:max)} END {for (i=1; i<=max; i++) {for (j=1; j<=NR; j++) printf "%s%s", a[i,j], (j==NR?RS:FS) }}' | sed 's/ /\t/g' > glup
mv glup $sample.LoadStats
```



## Run LoadStats.sh for each sample and then parse the output files
```{bash}
# make the appropriate counts
for file in *snpeff.minDP10.homD.EffList.gz ; do echo $file ; bash LoadStats.sh $file ; done

# calculate the relevant ratios: missense/syn and lof/syn
for file in *minDP10.homD.LoadStats ; do echo $file ; awk 'NR>1 { $5=$2/$1 ; $6=$3/$1} {print}' $file ; done | grep -v synonymous | paste - - | sed 's/.LoadStats//g' | sed 's/ /\t/g' 
```

Transfer the relevant counts and ratios to the TableS1_v6 for visualization.



## Visualize results

Load required libraries
```{r message=FALSE, warning=FALSE, paged.print=FALSE}
library(ggplot2)
library(cowplot)
library(ggrepel)
library(reshape2)
library(dplyr)
library(tidyverse)
library(data.table)
require(scales)
library(ggpubr)
library(tidyr)
library(PairedData)
library(gridExtra)
```


Open input data
```{r}
load = read.csv("WR_TableS1_v6.csv", h=T, fill=T, stringsAsFactors=T, fileEncoding="latin1")
```


```{r}
loadswr = subset(load, subspecies == "simum" & depth_of_coverage >= 10)
dim(loadswr)
```


```{r}
load$time_window=factor(load$time_window,levels=c("NWRpre", "NWRpost" ,"SWRpre", "SWRpost"))
loadss$time_window=factor(loadss$time_window,levels=c("NWRpre", "SWRpre", "SWRpost"))
```


```{r}
# missense to syn
SWRdnds = ggplot(loadswr, aes(time_window, homD_dNdS)) + theme_bw() +
  geom_boxplot(aes(colour=time_window), fill="gray70", lwd=.5, alpha=.4) + 
  geom_jitter(aes(colour=time_window), shape=17, position=position_jitter(0.11), cex=3, alpha=.6) + 
  
  scale_y_continuous(limits=c(0.7, 0.85), breaks=seq(0.7, 0.85, 0.02)) +
  
  scale_color_manual(values= (c("royalblue4", "steelblue3"))) +
  stat_compare_means(paired=F, method = "wilcox.test", comparisons = list(c("SWRpost", "SWRpre")), method.args = list(alternative = "less")) +
  theme(legend.position = "none") +
  geom_text_repel(data=subset(loadswr, time_window=="SWRpre"), aes(label=sample_name), size=3) +
  xlab(NULL) + ylab("Homozygous derived missense/synonymous")

SWRdnds

# LoF to syn
SWRlofsyn = ggplot(loadswr, aes(time_window, homD_lofsyn)) + theme_bw() +
  geom_boxplot(aes(colour=time_window), fill="gray70", lwd=.5, alpha=.4) + 
  geom_jitter(aes(colour=time_window), shape=17, position=position_jitter(0.11), cex=3, alpha=.6) + 
  
  scale_y_continuous(limits=c(0.0135, 0.0265), breaks=seq(0.014, 0.026, 0.002)) +
  scale_color_manual(values= (c("royalblue4", "steelblue3"))) +
  
  stat_compare_means(paired=F, method = "wilcox.test", comparisons = list(c("SWRpre", "SWRpost")), method.args = list(alternative = "greater")) +
  theme(legend.position = "none") +
   geom_text_repel(data=subset(loadswr, time_window=="SWRpre"), aes(label=sample_name), size=3) +
  xlab(NULL) + ylab("Homozygous derived LoF/synonymous")

SWRlofsyn
```

```{r}
pdf("WR_v5_SWRhomDLoadMinDP10.pdf", useDingbats = F)
ggarrange(SWRdnds, SWRlofsyn, nrow=2 )
dev.off()
```
































